Fix Ratva for abhyasa

tbdasap opened the pull request:
Added code and test cases to include णत्व for abhyasa

neeleshb commented:
All these (with णत्व) are correct forms. This has been elaborated upon in some commentaries of Paribhashendushekhara. However I could not find any forms directly given as examples in Ashtadhyayi commentaries.  Only one reference is seen in Tattavabodhini - "एवं च प्रणिनायेत्यादौ द्वितीयस्य णत्वाऽभावः सिद्धः" in TB, 6.1.3

tbdasap commented on vidyut-prakriya/tests/integration/kashika_8_4.rs:
```diff
@@ -86,6 +86,19 @@ fn sutra_8_4_14() {
     assert_has_krdanta(&["pra"], &nard, Krt::Rvul, &["pranardaka"]);
+
+    // These are for testing with Dvitvam with Upsarga
+    let ni = d("RI\\Y", Bhvadi);
```
> I prefer that all of the `kashika` tests come directly from the KV. Instead, we can put these in `regressions.rs` as `fn natva_8_4_14` or similar.

Thank you. That is good guidance for this as well as the prior PR.

tbdasap replied:
Done

tbdasap commented:
This turned out to be a much larger effort. Briefly
- refactored some dhatu prep steps
- Added tests for reasonable णत्व coverage
   - fixes like निर्गमन परिभवनीय etc.
- Fixed some rule numbers i.e. the processing did not match to the correct rule.
- Fixed some upasarga related logic for ष्ठा i.e परा ( अभि etc.) + ष्ठा decline परस्मैपदी *only*

tbdasap commented on vidyut-prakriya/src/core/prakriya.rs:
```diff
@@ -199,8 +201,10 @@ impl Config {
 #[derive(Clone, Default, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub(crate) enum Stage {
     #[default]
+    DhatuPrep,
```
Could you clarify in the above context ? Presuming this was about the next clone..

tbdasap commented on vidyut-prakriya/src/core/term.rs:
```diff
@@ -225,6 +225,10 @@ impl Term {
     }
 
+    pub fn u_adi(&self) -> Option<char> {
+        self.u.clone().unwrap().bytes().next().map(|x| x as char)
```
Thank you .. Fixed

tbdasap commented on vidyut-prakriya/src/samjna.rs:
```diff
@@ -448,11 +450,14 @@ pub fn try_decide_pratipadika(p: &mut Prakriya) -> Option<()> {
             && !t.is_agama()
             && !t.is_abhyasa()
             && !t.is_pada()
+            && !t.is_upasarga()
```
Good catch! fixed.. 
upasarga  --> nipata (1.4.58??). Nipata is pratipadika as per Kashika/Mahabhashya 1.2.45

tbdasap commented on vidyut-prakriya/tests/integration/kashika_3_2.rs:
```diff
@@ -526,6 +523,16 @@ fn sutra_3_2_45() {
 
+#[test]
+fn sutra_3_2_46() {
+    let mut myd = d("Df\\Y", Bhvadi);
```
Fixed

tbdasap commented on vidyut-prakriya/tests/integration/kashika_6_1.rs:
```diff
@@ -524,12 +524,8 @@ fn sutra_6_1_48() {
-        Lat,
-        &["sADayati", "seDayati"],
-    );
+    let zid = d("zi\\Du~", Divadi);
```
Fixed

tbdasap commented on vidyut-prakriya/tests/integration/regressions.rs:
```diff
@@ -336,3 +336,33 @@ fn apombhyat() {
+    assert_has_krdanta(&[], &san(&myd), Krt::SAnac, &["tizWAsamAna"]);
+    // "pari" upasarga is always parasmaipadi
+    assert_has_krdanta(&["pari"], &san(&myd), Krt::Satf, &["paritizWAsat"]);
+    assert_has_krdanta(&["pari"], &san(&myd), Krt::SAnac, &[]); // fails
```
Done

tbdasap commented on vidyut-prakriya/tests/integration/regressions.rs:
```diff
@@ -336,3 +336,33 @@ fn apombhyat() {
+    assert_has_krdanta(&["pari"], &san(&myd), Krt::SAnac, &[]); // fails
+    // "pra" upasarga is always atmanepadi
+    assert_has_krdanta(&["pra"], &san(&myd), Krt::SAnac, &["pratizWAsamAna"]);
+    assert_has_krdanta(&["pra"], &san(&myd), Krt::Satf, &[]); // fails
```
Done

tbdasap commented on vidyut-prakriya/src/it_agama.rs:
```diff
@@ -502,12 +503,16 @@ fn run_valadau_ardhadhatuke_before_attva_for_term(ip: &mut ItPrakriya) -> Option
             }
-            if can_run {
+            if can_block {
+                // Block iw-agama
```
Done!

tbdasap commented on vidyut-prakriya/src/tripadi/pada_8_4.rs:
```diff
@@ -153,17 +207,45 @@ fn try_natva_for_span(
+                if dhatu.has_u_adi(IC) {
+                    ip.p.run_at("8.4.32", i_y, |t| t.find_and_replace_text("n", "R"));
+                } else {
+                    // Do nothing but note the step for clarification
```
Done!

tbdasap commented on vidyut-prakriya/tests/integration/kashika_8_4.rs:
```diff
@@ -73,6 +73,42 @@ fn sutra_8_4_2() {
+    // 8.4.8 has two rules
+    //  Nipatana ufor vAhana always
+    // 8.4.8.1 : Ahita for vAhanam
+    rule_choices.push(RuleChoice {
```
Done

tbdasap commented on vidyut-prakriya/src/tripadi/pada_8_4.rs:
```diff
@@ -61,6 +63,20 @@ fn try_natva_for_span(
         }
     }
 
+    /* This is only for debugging
```
Done!

tbdasap commented on vidyut-prakriya/src/atidesha.rs:
```diff
@@ -286,6 +285,9 @@ fn run_before_attva_for_index(p: &mut Prakriya, i: usize) -> Option<()> {
     }
 }
 
+/* Sriram:: Refactoring seems to have fixed this circular dep.
```
And remove this function also ?

tbdasap replied:
Done

tbdasap commented on vidyut-prakriya/src/ashtadhyayi.rs:
```diff
@@ -177,9 +177,9 @@ fn prepare_dhatu(p: &mut Prakriya, dhatu: &Dhatu, args: MainArgs) -> Result<()>
-
     p.debug("~~~~~~~~~~~~~~ <prepare-dhatu> ~~~~~~~~~~~~~~~~~~");
-
+    let orig_stage = p.stage.clone();
```
Thank you! will fix this and the next one

tbdasap replied:
Done.. had to add "Copy" for Enum.

tbdasap commented:
> Added some more small comments.
> 
> My one high-level question is about the use and necessity of DhatuPrep. Clearly it solves problems, but it's been a while since I've thought about the code or the Ashtadhyayi in detail so I would love some help understanding why this is necessary, whether this is the traditional approach that vaiyAkaraNas use, etc. Basically, I would appreciate as much context as possible on this decision to add a new stage.
>
> I ask because stages also change the execution model of the code, since one must keep in mind not only where the prakriya is in the code but what stage it has. Stage exists as it is now because it solves certain problems I wasn't sure how to solve otherwise, but I want to expand it only if it is really necessary.

TLD;R this has
- more to do refactoring for clarity
- little to do with "traditional approaches" (I've only rudimentary understanding of such traditions).
- has no intention of creating an official `stage` in the Prakriya.. just a convenience.

I should acknowledge that for someone who learnt Rust _after checking_ out this codebase, this has been a joy to work with.


**What refactoring**
It was related to  [run_main_rules](https://github.com/ambuda-org/vidyut/blob/main/vidyut-prakriya/src/ashtadhyayi.rs#L595-L596) being invoked deep inside `prepare_dhatu` as well as right after it. 
I tried to iteratively tease it apart and just used enough in `run_prepare_dhatu_rules` to make it work inside `prepare_dhatu`.  Eventually there were parts that needed to know whether something was 
to be invoked in the "prepare_dhatu_inner" context or not.
 
The prakriya `stage structure` was a _convenient place_  to plug it in ... perhaps another field would have worked too . After many code iterations it was possible to get some sort  of minimal set and get all the unittests to pass. Some of the "failing" tests became more amenable to fixing.

As of now there are really 3 instances of this usage
- angasya.rs  (2 instances)
- ac_sandhi.rs ( 1 instance)
where the check is needed for the context (i.e. run_prepare_dhatu_rules or not)

1. I view the above as trackers for those areas of code that need some refactoring but were traded-off as acceptable hack for more accurate functionality.
2. An **unintended** side-effect was that certain `functions` became vestigial . Your intuition was accurate when annotating them with `hacky` comments/names and it was quite satisfying and validating (for me)  to see them precipitated out.

**As to the Why**
As to the motivation for teasing out well, while trying to modify code/ edit/debug. I observed that it wasn't clear that 
- after a sutra rule was applied.
- and a subsequent rule "unapplied it" (perhaps an apavaada or something).
- how all of this was tracked and done correctly if for eg. `run_make_rules` was run repeatedly.

Here is a very specific example
- [6.4.92](https://ashtadhyayi.com/sutraani/6/4/92) मितां ह्रस्वः
- [6.4.93](https://ashtadhyayi.com/sutraani/6/4/93) चिण्णमुलोर्दीर्घोऽन्यतरस्याम्

My understanding is that
1. they should be applied only once in the prakriya for a Term (Dhatu) 
2. there should be a guarantee that 6.4.92 will not be applied after 6.4.93

Prakriya for [णिजन्ते - भावकर्मणोः लुङ्लकारः (आत्मनेपदम्)  प्र॰पु॰  एक॰](https://ashtadhyayi.com/dhatu/10.0124?tab=nich_yak) is snapshotted. Here 
- 6.4.92 is applied thrice on चि
- अचापि and अचायि could perhaps be valid outputs.
 
 ![prakriya for अचयि ](https://github.com/user-attachments/assets/2cdc6e7c-a625-479b-9cbb-e5e65af8aa26)

 Hope this helps

tbdasap commented on vidyut-prakriya/src/ashtadhyayi.rs:
```diff
@@ -211,13 +212,13 @@ fn prepare_dhatu_inner(p: &mut Prakriya, dhatu: &Dhatu, args: MainArgs) -> Resul
     p.debug("~~~~~~~~~~~~~~ </prepare-dhatu> ~~~~~~~~~~~~~~~~~~");
-
-    // Defer tripadi until we add other pratyayas.
+    p.stage = orig_stage.clone();
```
Aah! yes... it is an enum. so need to copy!!

tbdasap replied:
Done

tbdasap commented:
> I also see the Makefile now has a merge conflict, so please rebase when you have a chance.
👍  for the reminder. The build seems to be broken. I will try and look at what the reason might be.

Here is what I will be doing next and it is also going to take a week or more for me to get to it.
1. Run all the tinnantas/kridantas and document the diffs for review
2. Some of the prakriyas will have changed. That will need to figuring out building/testing of the python bindings

tbdasap commented:
The wasm and python-bindings seem to suffer from the `release` and `non-release` build flavors
 behaving differently in a _subtle way_ resulting in failure of the detailed prakriya test (in `bindings-python` for `test/unit/prakriya` for `भवति`)

- To support the observation for wasm `vidyut-prakriya/scripts/run_debugger.sh` has updates for simultaneously running both release (port 8001) and non-release (port 8000) versions so both outputs may be _visually compared_. The color differences for `Terms` will be different although the prakriya rule sequences **_are identical._**
- `bindings-python/Makefile` has been modified to make it work so that if `make test` succeeds, the `gh-runner` should succeed. 
   - `make test` implementation _requires_ to use `maturin develop --release` (for prakriya test success with gh-runner)
   - `make test RELEASE=` will fail (`non-release` build)
- `make integration_tests` fails as below. Perhaps this is WIP ??
```
FAILED test/integration/test_cheda.py::test_run_for_word[devaH] - AssertionError: assert None is not None
FAILED test/integration/test_cheda.py::test_run_for_phrase[gacCatyarjunaH-expected0] - AssertionError: assert ['gacCati', 'arjunas'] == ['gacCatyarjunaH']
FAILED test/integration/test_prakriya.py::test_dhatupatha_ganasutras - assert 19 == 18
```

tbdasap commented:
Regression changes across all tinnantas and krdantas.

#### Tinantas
|Derivation for | After | Before |
|---------------|---------|-------|
| ciY, curAdi, 124, , Karmani, Lun, Prathama, Eka | अचापि , अचायि , अचपि , अचयि | अचायि , अचपि , अचयि
| aja~, BvAdi, 262, san, Kartari, Lit, Prathama, Eka | अजिजिषाञ्चकार , अजिजिषामास , अजिजिषाम्बभूव , विवीषाञ्चकार , विवीषामास , विवीषाम्बभूव | अजिवयिषाञ्चकार , अजिवयिषामास , अजिवयिषाम्बभूव , विवीषाञ्चकार , विवीषामास , विवीषाम्बभूव
| aja~, BvAdi, 262, san, Kartari, Lit, Prathama, Dvi | अजिजिषाञ्चक्रतुः , अजिजिषामासतुः , अजिजिषाम्बभूवतुः , विवीषाञ्चक्रतुः , विवीषामासतुः , विवीषाम्बभूवतुः | अजिवयिषाञ्चक्रतुः , अजिवयिषामासतुः , अजिवयिषाम्बभूवतुः , विवीषाञ्चक्रतुः , विवीषामासतुः , विवीषाम्बभूवतुः
| aja~, BvAdi, 262, san, Kartari, Lit, Prathama, Bahu | अजिजिषाञ्चक्रुः , अजिजिषामासुः , अजिजिषाम्बभूवुः , विवीषाञ्चक्रुः , विवीषामासुः , विवीषाम्बभूवुः | अजिवयिषाञ्चक्रुः , अजिवयिषामासुः , अजिवयिषाम्बभूवुः , विवीषाञ्चक्रुः , विवीषामासुः , विवीषाम्बभूवुः
| aja~, BvAdi, 262, san, Kartari, Lit, Madhyama, Eka | अजिजिषाञ्चकर्थ , अजिजिषामासिथ , अजिजिषाम्बभूविथ , विवीषाञ्चकर्थ , विवीषामासिथ , विवीषाम्बभूविथ | अजिवयिषाञ्चकर्थ , अजिवयिषामासिथ , अजिवयिषाम्बभूविथ , विवीषाञ्चकर्थ , विवीषामासिथ , विवीषाम्बभूविथ
| aja~, BvAdi, 262, san, Kartari, Lit, Madhyama, Dvi | अजिजिषाञ्चक्रथुः , अजिजिषामासथुः , अजिजिषाम्बभूवथुः , विवीषाञ्चक्रथुः , विवीषामासथुः , विवीषाम्बभूवथुः | अजिवयिषाञ्चक्रथुः , अजिवयिषामासथुः , अजिवयिषाम्बभूवथुः , विवीषाञ्चक्रथुः , विवीषामासथुः , विवीषाम्बभूवथुः
| aja~, BvAdi, 262, san, Kartari, Lit, Madhyama, Bahu | अजिजिषाञ्चक्र , अजिजिषामास , अजिजिषाम्बभूव , विवीषाञ्चक्र , विवीषामास , विवीषाम्बभूव | अजिवयिषाञ्चक्र , अजिवयिषामास , अजिवयिषाम्बभूव , विवीषाञ्चक्र , विवीषामास , विवीषाम्बभूव
| aja~, BvAdi, 262, san, Kartari, Lit, Uttama, Eka | अजिजिषाञ्चकार , अजिजिषाञ्चकर , अजिजिषामास , अजिजिषाम्बभूव , विवीषाञ्चकार , विवीषाञ्चकर , विवीषामास , विवीषाम्बभूव | अजिवयिषाञ्चकार , अजिवयिषाञ्चकर , अजिवयिषामास , अजिवयिषाम्बभूव , विवीषाञ्चकार , विवीषाञ्चकर , विवीषामास , विवीषाम्बभूव
| aja~, BvAdi, 262, san, Kartari, Lit, Uttama, Dvi | अजिजिषाञ्चकृव , अजिजिषामासिव , अजिजिषाम्बभूविव , विवीषाञ्चकृव , विवीषामासिव , विवीषाम्बभूविव | अजिवयिषाञ्चकृव , अजिवयिषामासिव , अजिवयिषाम्बभूविव , विवीषाञ्चकृव , विवीषामासिव , विवीषाम्बभूविव
| aja~, BvAdi, 262, san, Kartari, Lit, Uttama, Bahu | अजिजिषाञ्चकृम , अजिजिषामासिम , अजिजिषाम्बभूविम , विवीषाञ्चकृम , विवीषामासिम , विवीषाम्बभूविम | अजिवयिषाञ्चकृम , अजिवयिषामासिम , अजिवयिषाम्बभूविम , विवीषाञ्चकृम , विवीषामासिम , विवीषाम्बभूविम
| aja~, BvAdi, 262, san, Kartari, Lut, Prathama, Eka | अजिजिषिता , विवीषिता | अजिवयिषिता , विवीषिता
| aja~, BvAdi, 262, san, Kartari, Lut, Prathama, Dvi | अजिजिषितारौ , विवीषितारौ | अजिवयिषितारौ , विवीषितारौ
| aja~, BvAdi, 262, san, Kartari, Lut, Prathama, Bahu | अजिजिषितारः , विवीषितारः | अजिवयिषितारः , विवीषितारः
| aja~, BvAdi, 262, san, Kartari, Lut, Madhyama, Eka | अजिजिषितासि , विवीषितासि | अजिवयिषितासि , विवीषितासि
| aja~, BvAdi, 262, san, Kartari, Lut, Madhyama, Dvi | अजिजिषितास्थः , विवीषितास्थः | अजिवयिषितास्थः , विवीषितास्थः
| aja~, BvAdi, 262, san, Kartari, Lut, Madhyama, Bahu | अजिजिषितास्थ , विवीषितास्थ | अजिवयिषितास्थ , विवीषितास्थ
| aja~, BvAdi, 262, san, Kartari, Lut, Uttama, Eka | अजिजिषितास्मि , विवीषितास्मि | अजिवयिषितास्मि , विवीषितास्मि
| aja~, BvAdi, 262, san, Kartari, Lut, Uttama, Dvi | अजिजिषितास्वः , विवीषितास्वः | अजिवयिषितास्वः , विवीषितास्वः
| aja~, BvAdi, 262, san, Kartari, Lut, Uttama, Bahu | अजिजिषितास्मः , विवीषितास्मः | अजिवयिषितास्मः , विवीषितास्मः
| aja~, BvAdi, 262, san, Kartari, Lrt, Prathama, Eka | अजिजिषिष्यति , विवीषिष्यति | अजिवयिषिष्यति , विवीषिष्यति
| aja~, BvAdi, 262, san, Kartari, Lrt, Prathama, Dvi | अजिजिषिष्यतः , विवीषिष्यतः | अजिवयिषिष्यतः , विवीषिष्यतः
| aja~, BvAdi, 262, san, Kartari, Lrt, Prathama, Bahu | अजिजिषिष्यन्ति , विवीषिष्यन्ति | अजिवयिषिष्यन्ति , विवीषिष्यन्ति
| aja~, BvAdi, 262, san, Kartari, Lrt, Madhyama, Eka | अजिजिषिष्यसि , विवीषिष्यसि | अजिवयिषिष्यसि , विवीषिष्यसि
| aja~, BvAdi, 262, san, Kartari, Lrt, Madhyama, Dvi | अजिजिषिष्यथः , विवीषिष्यथः | अजिवयिषिष्यथः , विवीषिष्यथः
| aja~, BvAdi, 262, san, Kartari, Lrt, Madhyama, Bahu | अजिजिषिष्यथ , विवीषिष्यथ | अजिवयिषिष्यथ , विवीषिष्यथ
| aja~, BvAdi, 262, san, Kartari, Lrt, Uttama, Eka | अजिजिषिष्यामि , विवीषिष्यामि | अजिवयिषिष्यामि , विवीषिष्यामि
| aja~, BvAdi, 262, san, Kartari, Lrt, Uttama, Dvi | अजिजिषिष्यावः , विवीषिष्यावः | अजिवयिषिष्यावः , विवीषिष्यावः
| aja~, BvAdi, 262, san, Kartari, Lrt, Uttama, Bahu | अजिजिषिष्यामः , विवीषिष्यामः | अजिवयिषिष्यामः , विवीषिष्यामः
| aja~, BvAdi, 262, san, Kartari, Let, Prathama, Eka | अजिजिषाति , विवीषाति | अजिवयिषाति , विवीषाति
| aja~, BvAdi, 262, san, Kartari, Let, Prathama, Dvi | अजिजिषातः , विवीषातः | अजिवयिषातः , विवीषातः
| aja~, BvAdi, 262, san, Kartari, Let, Prathama, Bahu | अजिजिषान्ति , विवीषान्ति | अजिवयिषान्ति , विवीषान्ति
| aja~, BvAdi, 262, san, Kartari, Let, Madhyama, Eka | अजिजिषासि , विवीषासि | अजिवयिषासि , विवीषासि
| aja~, BvAdi, 262, san, Kartari, Let, Madhyama, Dvi | अजिजिषाथः , विवीषाथः | अजिवयिषाथः , विवीषाथः
| aja~, BvAdi, 262, san, Kartari, Let, Madhyama, Bahu | अजिजिषाथ , विवीषाथ | अजिवयिषाथ , विवीषाथ
| aja~, BvAdi, 262, san, Kartari, Let, Uttama, Eka | अजिजिषामि , विवीषामि | अजिवयिषामि , विवीषामि
| aja~, BvAdi, 262, san, Kartari, Let, Uttama, Dvi | अजिजिषाव , अजिजिषावः , विवीषाव , विवीषावः | अजिवयिषाव , अजिवयिषावः , विवीषाव , विवीषावः
| aja~, BvAdi, 262, san, Kartari, Let, Uttama, Bahu | अजिजिषाम , अजिजिषामः , विवीषाम , विवीषामः | अजिवयिषाम , अजिवयिषामः , विवीषाम , विवीषामः
| aja~, BvAdi, 262, san, Kartari, Lun, Prathama, Eka | आजिजिषीद् , आजिजिषीत् , अविवीषीद् , अविवीषीत् | आजिवयिषीद् , आजिवयिषीत् , अविवीषीद् , अविवीषीत्
| aja~, BvAdi, 262, san, Kartari, Lun, Prathama, Dvi | आजिजिषिष्टाम् , अविवीषिष्टाम् | आजिवयिषिष्टाम् , अविवीषिष्टाम्
| aja~, BvAdi, 262, san, Kartari, Lun, Prathama, Bahu | आजिजिषिषुः , अविवीषिषुः | आजिवयिषिषुः , अविवीषिषुः
| aja~, BvAdi, 262, san, Kartari, Lun, Madhyama, Eka | आजिजिषीः , अविवीषीः | आजिवयिषीः , अविवीषीः
| aja~, BvAdi, 262, san, Kartari, Lun, Madhyama, Dvi | आजिजिषिष्टम् , अविवीषिष्टम् | आजिवयिषिष्टम् , अविवीषिष्टम्
| aja~, BvAdi, 262, san, Kartari, Lun, Madhyama, Bahu | आजिजिषिष्ट , अविवीषिष्ट | आजिवयिषिष्ट , अविवीषिष्ट
| aja~, BvAdi, 262, san, Kartari, Lun, Uttama, Eka | आजिजिषिषम् , अविवीषिषम् | आजिवयिषिषम् , अविवीषिषम्
| aja~, BvAdi, 262, san, Kartari, Lun, Uttama, Dvi | आजिजिषिष्व , अविवीषिष्व | आजिवयिषिष्व , अविवीषिष्व
| aja~, BvAdi, 262, san, Kartari, Lun, Uttama, Bahu | आजिजिषिष्म , अविवीषिष्म | आजिवयिषिष्म , अविवीषिष्म
| aja~, BvAdi, 262, san, Kartari, Lrn, Prathama, Eka | आजिजिषिष्यद् , आजिजिषिष्यत् , अविवीषिष्यद् , अविवीषिष्यत् | आजिवयिषिष्यद् , आजिवयिषिष्यत् , अविवीषिष्यद् , अविवीषिष्यत्
| aja~, BvAdi, 262, san, Kartari, Lrn, Prathama, Dvi | आजिजिषिष्यताम् , अविवीषिष्यताम् | आजिवयिषिष्यताम् , अविवीषिष्यताम्
| aja~, BvAdi, 262, san, Kartari, Lrn, Prathama, Bahu | आजिजिषिष्यन् , अविवीषिष्यन् | आजिवयिषिष्यन् , अविवीषिष्यन्
| aja~, BvAdi, 262, san, Kartari, Lrn, Madhyama, Eka | आजिजिषिष्यः , अविवीषिष्यः | आजिवयिषिष्यः , अविवीषिष्यः
| aja~, BvAdi, 262, san, Kartari, Lrn, Madhyama, Dvi | आजिजिषिष्यतम् , अविवीषिष्यतम् | आजिवयिषिष्यतम् , अविवीषिष्यतम्
| aja~, BvAdi, 262, san, Kartari, Lrn, Madhyama, Bahu | आजिजिषिष्यत , अविवीषिष्यत | आजिवयिषिष्यत , अविवीषिष्यत
| aja~, BvAdi, 262, san, Kartari, Lrn, Uttama, Eka | आजिजिषिष्यम् , अविवीषिष्यम् | आजिवयिषिष्यम् , अविवीषिष्यम्
| aja~, BvAdi, 262, san, Kartari, Lrn, Uttama, Dvi | आजिजिषिष्याव , अविवीषिष्याव | आजिवयिषिष्याव , अविवीषिष्याव
| aja~, BvAdi, 262, san, Kartari, Lrn, Uttama, Bahu | आजिजिषिष्याम , अविवीषिष्याम | आजिवयिषिष्याम , अविवीषिष्याम
| aja~, BvAdi, 262, san, Karmani, Lat, Prathama, Eka | अजिजिष्यते , विवीष्यते | अजिवयिष्यते , विवीष्यते
| aja~, BvAdi, 262, san, Karmani, Lat, Prathama, Dvi | अजिजिष्येते , विवीष्येते | अजिवयिष्येते , विवीष्येते
| aja~, BvAdi, 262, san, Karmani, Lat, Prathama, Bahu | अजिजिष्यन्ते , विवीष्यन्ते | अजिवयिष्यन्ते , विवीष्यन्ते
| aja~, BvAdi, 262, san, Karmani, Lat, Madhyama, Eka | अजिजिष्यसे , विवीष्यसे | अजिवयिष्यसे , विवीष्यसे
| aja~, BvAdi, 262, san, Karmani, Lat, Madhyama, Dvi | अजिजिष्येथे , विवीष्येथे | अजिवयिष्येथे , विवीष्येथे
| aja~, BvAdi, 262, san, Karmani, Lat, Madhyama, Bahu | अजिजिष्यध्वे , विवीष्यध्वे | अजिवयिष्यध्वे , विवीष्यध्वे
| aja~, BvAdi, 262, san, Karmani, Lat, Uttama, Eka | अजिजिष्ये , विवीष्ये | अजिवयिष्ये , विवीष्ये
| aja~, BvAdi, 262, san, Karmani, Lat, Uttama, Dvi | अजिजिष्यावहे , विवीष्यावहे | अजिवयिष्यावहे , विवीष्यावहे
| aja~, BvAdi, 262, san, Karmani, Lat, Uttama, Bahu | अजिजिष्यामहे , विवीष्यामहे | अजिवयिष्यामहे , विवीष्यामहे
| aja~, BvAdi, 262, san, Karmani, Lit, Prathama, Eka | अजिजिषाञ्चक्रे , अजिजिषामासे , अजिजिषाम्बभूवे , विवीषाञ्चक्रे , विवीषामासे , विवीषाम्बभूवे | अजिवयिषाञ्चक्रे , अजिवयिषामासे , अजिवयिषाम्बभूवे , विवीषाञ्चक्रे , विवीषामासे , विवीषाम्बभूवे
| aja~, BvAdi, 262, san, Karmani, Lit, Prathama, Dvi | अजिजिषाञ्चक्राते , अजिजिषामासाते , अजिजिषाम्बभूवाते , विवीषाञ्चक्राते , विवीषामासाते , विवीषाम्बभूवाते | अजिवयिषाञ्चक्राते , अजिवयिषामासाते , अजिवयिषाम्बभूवाते , विवीषाञ्चक्राते , विवीषामासाते , विवीषाम्बभूवाते
| aja~, BvAdi, 262, san, Karmani, Lit, Prathama, Bahu | अजिजिषाञ्चक्रिरे , अजिजिषामासिरे , अजिजिषाम्बभूविरे , विवीषाञ्चक्रिरे , विवीषामासिरे , विवीषाम्बभूविरे | अजिवयिषाञ्चक्रिरे , अजिवयिषामासिरे , अजिवयिषाम्बभूविरे , विवीषाञ्चक्रिरे , विवीषामासिरे , विवीषाम्बभूविरे
| aja~, BvAdi, 262, san, Karmani, Lit, Madhyama, Eka | अजिजिषाञ्चकृषे , अजिजिषामासिषे , अजिजिषाम्बभूविषे , विवीषाञ्चकृषे , विवीषामासिषे , विवीषाम्बभूविषे | अजिवयिषाञ्चकृषे , अजिवयिषामासिषे , अजिवयिषाम्बभूविषे , विवीषाञ्चकृषे , विवीषामासिषे , विवीषाम्बभूविषे
| aja~, BvAdi, 262, san, Karmani, Lit, Madhyama, Dvi | अजिजिषाञ्चक्राथे , अजिजिषामासाथे , अजिजिषाम्बभूवाथे , विवीषाञ्चक्राथे , विवीषामासाथे , विवीषाम्बभूवाथे | अजिवयिषाञ्चक्राथे , अजिवयिषामासाथे , अजिवयिषाम्बभूवाथे , विवीषाञ्चक्राथे , विवीषामासाथे , विवीषाम्बभूवाथे
| aja~, BvAdi, 262, san, Karmani, Lit, Madhyama, Bahu | अजिजिषाञ्चकृढ्वे , अजिजिषामासिध्वे , अजिजिषाम्बभूविढ्वे , विवीषाञ्चकृढ्वे , विवीषामासिध्वे , विवीषाम्बभूविढ्वे | अजिवयिषाञ्चकृढ्वे , अजिवयिषामासिध्वे , अजिवयिषाम्बभूविढ्वे , विवीषाञ्चकृढ्वे , विवीषामासिध्वे , विवीषाम्बभूविढ्वे
| aja~, BvAdi, 262, san, Karmani, Lit, Uttama, Eka | अजिजिषाञ्चक्रे , अजिजिषामाहे , अजिजिषाम्बभूवे , विवीषाञ्चक्रे , विवीषामाहे , विवीषाम्बभूवे | अजिवयिषाञ्चक्रे , अजिवयिषामाहे , अजिवयिषाम्बभूवे , विवीषाञ्चक्रे , विवीषामाहे , विवीषाम्बभूवे
| aja~, BvAdi, 262, san, Karmani, Lit, Uttama, Dvi | अजिजिषाञ्चकृवहे , अजिजिषामासिवहे , अजिजिषाम्बभूविवहे , विवीषाञ्चकृवहे , विवीषामासिवहे , विवीषाम्बभूविवहे | अजिवयिषाञ्चकृवहे , अजिवयिषामासिवहे , अजिवयिषाम्बभूविवहे , विवीषाञ्चकृवहे , विवीषामासिवहे , विवीषाम्बभूविवहे
| aja~, BvAdi, 262, san, Karmani, Lit, Uttama, Bahu | अजिजिषाञ्चकृमहे , अजिजिषामासिमहे , अजिजिषाम्बभूविमहे , विवीषाञ्चकृमहे , विवीषामासिमहे , विवीषाम्बभूविमहे | अजिवयिषाञ्चकृमहे , अजिवयिषामासिमहे , अजिवयिषाम्बभूविमहे , विवीषाञ्चकृमहे , विवीषामासिमहे , विवीषाम्बभूविमहे
| aja~, BvAdi, 262, san, Karmani, Lut, Prathama, Eka | अजिजिषिता , विवीषिता | अजिवयिषिता , विवीषिता
| aja~, BvAdi, 262, san, Karmani, Lut, Prathama, Dvi | अजिजिषितारौ , विवीषितारौ | अजिवयिषितारौ , विवीषितारौ
| aja~, BvAdi, 262, san, Karmani, Lut, Prathama, Bahu | अजिजिषितारः , विवीषितारः | अजिवयिषितारः , विवीषितारः
| aja~, BvAdi, 262, san, Karmani, Lut, Madhyama, Eka | अजिजिषितासे , विवीषितासे | अजिवयिषितासे , विवीषितासे
| aja~, BvAdi, 262, san, Karmani, Lut, Madhyama, Dvi | अजिजिषितासाथे , विवीषितासाथे | अजिवयिषितासाथे , विवीषितासाथे
| aja~, BvAdi, 262, san, Karmani, Lut, Madhyama, Bahu | अजिजिषिताध्वे , विवीषिताध्वे | अजिवयिषिताध्वे , विवीषिताध्वे
| aja~, BvAdi, 262, san, Karmani, Lut, Uttama, Eka | अजिजिषिताहे , विवीषिताहे | अजिवयिषिताहे , विवीषिताहे
| aja~, BvAdi, 262, san, Karmani, Lut, Uttama, Dvi | अजिजिषितास्वहे , विवीषितास्वहे | अजिवयिषितास्वहे , विवीषितास्वहे
| aja~, BvAdi, 262, san, Karmani, Lut, Uttama, Bahu | अजिजिषितास्महे , विवीषितास्महे | अजिवयिषितास्महे , विवीषितास्महे
| aja~, BvAdi, 262, san, Karmani, Lrt, Prathama, Eka | अजिजिषिष्यते , विवीषिष्यते | अजिवयिषिष्यते , विवीषिष्यते
| aja~, BvAdi, 262, san, Karmani, Lrt, Prathama, Dvi | अजिजिषिष्येते , विवीषिष्येते | अजिवयिषिष्येते , विवीषिष्येते
| aja~, BvAdi, 262, san, Karmani, Lrt, Prathama, Bahu | अजिजिषिष्यन्ते , विवीषिष्यन्ते | अजिवयिषिष्यन्ते , विवीषिष्यन्ते
| aja~, BvAdi, 262, san, Karmani, Lrt, Madhyama, Eka | अजिजिषिष्यसे , विवीषिष्यसे | अजिवयिषिष्यसे , विवीषिष्यसे
| aja~, BvAdi, 262, san, Karmani, Lrt, Madhyama, Dvi | अजिजिषिष्येथे , विवीषिष्येथे | अजिवयिषिष्येथे , विवीषिष्येथे
| aja~, BvAdi, 262, san, Karmani, Lrt, Madhyama, Bahu | अजिजिषिष्यध्वे , विवीषिष्यध्वे | अजिवयिषिष्यध्वे , विवीषिष्यध्वे
| aja~, BvAdi, 262, san, Karmani, Lrt, Uttama, Eka | अजिजिषिष्ये , विवीषिष्ये | अजिवयिषिष्ये , विवीषिष्ये
| aja~, BvAdi, 262, san, Karmani, Lrt, Uttama, Dvi | अजिजिषिष्यावहे , विवीषिष्यावहे | अजिवयिषिष्यावहे , विवीषिष्यावहे
| aja~, BvAdi, 262, san, Karmani, Lrt, Uttama, Bahu | अजिजिषिष्यामहे , विवीषिष्यामहे | अजिवयिषिष्यामहे , विवीषिष्यामहे
| aja~, BvAdi, 262, san, Karmani, Let, Prathama, Eka | अजिजिष्याते , विवीष्याते | अजिवयिष्याते , विवीष्याते
| aja~, BvAdi, 262, san, Karmani, Let, Prathama, Dvi | अजिजिष्यैते , विवीष्यैते | अजिवयिष्यैते , विवीष्यैते
| aja~, BvAdi, 262, san, Karmani, Let, Prathama, Bahu | अजिजिष्याते , विवीष्याते | अजिवयिष्याते , विवीष्याते
| aja~, BvAdi, 262, san, Karmani, Let, Madhyama, Eka | अजिजिष्यासे , विवीष्यासे | अजिवयिष्यासे , विवीष्यासे
| aja~, BvAdi, 262, san, Karmani, Let, Madhyama, Dvi | अजिजिष्यैथे , विवीष्यैथे | अजिवयिष्यैथे , विवीष्यैथे
| aja~, BvAdi, 262, san, Karmani, Let, Madhyama, Bahu | अजिजिष्याध्वे , विवीष्याध्वे | अजिवयिष्याध्वे , विवीष्याध्वे
| aja~, BvAdi, 262, san, Karmani, Let, Uttama, Eka | अजिजिष्यै , विवीष्यै | अजिवयिष्यै , विवीष्यै
| aja~, BvAdi, 262, san, Karmani, Let, Uttama, Dvi | अजिजिष्यावहे , विवीष्यावहे | अजिवयिष्यावहे , विवीष्यावहे
| aja~, BvAdi, 262, san, Karmani, Let, Uttama, Bahu | अजिजिष्यामहे , विवीष्यामहे | अजिवयिष्यामहे , विवीष्यामहे
| aja~, BvAdi, 262, san, Karmani, Lot, Prathama, Eka | अजिजिष्यताम् , विवीष्यताम् | अजिवयिष्यताम् , विवीष्यताम्
| aja~, BvAdi, 262, san, Karmani, Lot, Prathama, Dvi | अजिजिष्येताम् , विवीष्येताम् | अजिवयिष्येताम् , विवीष्येताम्
| aja~, BvAdi, 262, san, Karmani, Lot, Prathama, Bahu | अजिजिष्यन्ताम् , विवीष्यन्ताम् | अजिवयिष्यन्ताम् , विवीष्यन्ताम्
| aja~, BvAdi, 262, san, Karmani, Lot, Madhyama, Eka | अजिजिष्यस्व , विवीष्यस्व | अजिवयिष्यस्व , विवीष्यस्व
| aja~, BvAdi, 262, san, Karmani, Lot, Madhyama, Dvi | अजिजिष्येथाम् , विवीष्येथाम् | अजिवयिष्येथाम् , विवीष्येथाम्
| aja~, BvAdi, 262, san, Karmani, Lot, Madhyama, Bahu | अजिजिष्यध्वम् , विवीष्यध्वम् | अजिवयिष्यध्वम् , विवीष्यध्वम्
| aja~, BvAdi, 262, san, Karmani, Lot, Uttama, Eka | अजिजिष्यै , विवीष्यै | अजिवयिष्यै , विवीष्यै
| aja~, BvAdi, 262, san, Karmani, Lot, Uttama, Dvi | अजिजिष्यावहै , विवीष्यावहै | अजिवयिष्यावहै , विवीष्यावहै
| aja~, BvAdi, 262, san, Karmani, Lot, Uttama, Bahu | अजिजिष्यामहै , विवीष्यामहै | अजिवयिष्यामहै , विवीष्यामहै
| aja~, BvAdi, 262, san, Karmani, Lan, Prathama, Eka | आजिजिष्यत , अविवीष्यत | आजिवयिष्यत , अविवीष्यत
| aja~, BvAdi, 262, san, Karmani, Lan, Prathama, Dvi | आजिजिष्येताम् , अविवीष्येताम् | आजिवयिष्येताम् , अविवीष्येताम्
| aja~, BvAdi, 262, san, Karmani, Lan, Prathama, Bahu | आजिजिष्यन्त , अविवीष्यन्त | आजिवयिष्यन्त , अविवीष्यन्त
| aja~, BvAdi, 262, san, Karmani, Lan, Madhyama, Eka | आजिजिष्यथाः , अविवीष्यथाः | आजिवयिष्यथाः , अविवीष्यथाः
| aja~, BvAdi, 262, san, Karmani, Lan, Madhyama, Dvi | आजिजिष्येथाम् , अविवीष्येथाम् | आजिवयिष्येथाम् , अविवीष्येथाम्
| aja~, BvAdi, 262, san, Karmani, Lan, Madhyama, Bahu | आजिजिष्यध्वम् , अविवीष्यध्वम् | आजिवयिष्यध्वम् , अविवीष्यध्वम्
| aja~, BvAdi, 262, san, Karmani, Lan, Uttama, Eka | आजिजिष्ये , अविवीष्ये | आजिवयिष्ये , अविवीष्ये
| aja~, BvAdi, 262, san, Karmani, Lan, Uttama, Dvi | आजिजिष्यावहि , अविवीष्यावहि | आजिवयिष्यावहि , अविवीष्यावहि
| aja~, BvAdi, 262, san, Karmani, Lan, Uttama, Bahu | आजिजिष्यामहि , अविवीष्यामहि | आजिवयिष्यामहि , अविवीष्यामहि
| aja~, BvAdi, 262, san, Karmani, VidhiLin, Prathama, Eka | अजिजिष्येत , विवीष्येत | अजिवयिष्येत , विवीष्येत
| aja~, BvAdi, 262, san, Karmani, VidhiLin, Prathama, Dvi | अजिजिष्येयाताम् , विवीष्येयाताम् | अजिवयिष्येयाताम् , विवीष्येयाताम्
| aja~, BvAdi, 262, san, Karmani, VidhiLin, Prathama, Bahu | अजिजिष्येरन् , विवीष्येरन् | अजिवयिष्येरन् , विवीष्येरन्
| aja~, BvAdi, 262, san, Karmani, VidhiLin, Madhyama, Eka | अजिजिष्येथाः , विवीष्येथाः | अजिवयिष्येथाः , विवीष्येथाः
| aja~, BvAdi, 262, san, Karmani, VidhiLin, Madhyama, Dvi | अजिजिष्येयाथाम् , विवीष्येयाथाम् | अजिवयिष्येयाथाम् , विवीष्येयाथाम्
| aja~, BvAdi, 262, san, Karmani, VidhiLin, Madhyama, Bahu | अजिजिष्येध्वम् , विवीष्येध्वम् | अजिवयिष्येध्वम् , विवीष्येध्वम्
| aja~, BvAdi, 262, san, Karmani, VidhiLin, Uttama, Eka | अजिजिष्येय , विवीष्येय | अजिवयिष्येय , विवीष्येय
| aja~, BvAdi, 262, san, Karmani, VidhiLin, Uttama, Dvi | अजिजिष्येवहि , विवीष्येवहि | अजिवयिष्येवहि , विवीष्येवहि
| aja~, BvAdi, 262, san, Karmani, VidhiLin, Uttama, Bahu | अजिजिष्येमहि , विवीष्येमहि | अजिवयिष्येमहि , विवीष्येमहि
| aja~, BvAdi, 262, san, Karmani, AshirLin, Prathama, Eka | अजिजिषिषीष्ट , विवीषिषीष्ट | अजिवयिषिषीष्ट , विवीषिषीष्ट
| aja~, BvAdi, 262, san, Karmani, AshirLin, Prathama, Dvi | अजिजिषिषीयास्ताम् , विवीषिषीयास्ताम् | अजिवयिषिषीयास्ताम् , विवीषिषीयास्ताम्
| aja~, BvAdi, 262, san, Karmani, AshirLin, Prathama, Bahu | अजिजिषिषीरन् , विवीषिषीरन् | अजिवयिषिषीरन् , विवीषिषीरन्
| aja~, BvAdi, 262, san, Karmani, AshirLin, Madhyama, Eka | अजिजिषिषीष्ठाः , विवीषिषीष्ठाः | अजिवयिषिषीष्ठाः , विवीषिषीष्ठाः
| aja~, BvAdi, 262, san, Karmani, AshirLin, Madhyama, Dvi | अजिजिषिषीयास्थाम् , विवीषिषीयास्थाम् | अजिवयिषिषीयास्थाम् , विवीषिषीयास्थाम्
| aja~, BvAdi, 262, san, Karmani, AshirLin, Madhyama, Bahu | अजिजिषिषीध्वम् , विवीषिषीध्वम् | अजिवयिषिषीध्वम् , विवीषिषीध्वम्
| aja~, BvAdi, 262, san, Karmani, AshirLin, Uttama, Eka | अजिजिषिषीय , विवीषिषीय | अजिवयिषिषीय , विवीषिषीय
| aja~, BvAdi, 262, san, Karmani, AshirLin, Uttama, Dvi | अजिजिषिषीवहि , विवीषिषीवहि | अजिवयिषिषीवहि , विवीषिषीवहि
| aja~, BvAdi, 262, san, Karmani, AshirLin, Uttama, Bahu | अजिजिषिषीमहि , विवीषिषीमहि | अजिवयिषिषीमहि , विवीषिषीमहि
| aja~, BvAdi, 262, san, Karmani, Lun, Prathama, Eka | आजिजिषि , अविवीषि | आजिवयिषि , अविवीषि
| aja~, BvAdi, 262, san, Karmani, Lun, Prathama, Dvi | आजिजिषिषाताम् , अविवीषिषाताम् | आजिवयिषिषाताम् , अविवीषिषाताम्
| aja~, BvAdi, 262, san, Karmani, Lun, Prathama, Bahu | आजिजिषिषत , अविवीषिषत | आजिवयिषिषत , अविवीषिषत
| aja~, BvAdi, 262, san, Karmani, Lun, Madhyama, Eka | आजिजिषिष्ठाः , अविवीषिष्ठाः | आजिवयिषिष्ठाः , अविवीषिष्ठाः
| aja~, BvAdi, 262, san, Karmani, Lun, Madhyama, Dvi | आजिजिषिषाथाम् , अविवीषिषाथाम् | आजिवयिषिषाथाम् , अविवीषिषाथाम्
| aja~, BvAdi, 262, san, Karmani, Lun, Madhyama, Bahu | आजिजिषिध्वम् , अविवीषिध्वम् | आजिवयिषिध्वम् , अविवीषिध्वम्
| aja~, BvAdi, 262, san, Karmani, Lun, Uttama, Eka | आजिजिषिषि , अविवीषिषि | आजिवयिषिषि , अविवीषिषि
| aja~, BvAdi, 262, san, Karmani, Lun, Uttama, Dvi | आजिजिषिष्वहि , अविवीषिष्वहि | आजिवयिषिष्वहि , अविवीषिष्वहि
| aja~, BvAdi, 262, san, Karmani, Lun, Uttama, Bahu | आजिजिषिष्महि , अविवीषिष्महि | आजिवयिषिष्महि , अविवीषिष्महि
| aja~, BvAdi, 262, san, Karmani, Lrn, Prathama, Eka | आजिजिषिष्यत , अविवीषिष्यत | आजिवयिषिष्यत , अविवीषिष्यत
| aja~, BvAdi, 262, san, Karmani, Lrn, Prathama, Dvi | आजिजिषिष्येताम् , अविवीषिष्येताम् | आजिवयिषिष्येताम् , अविवीषिष्येताम्
| aja~, BvAdi, 262, san, Karmani, Lrn, Prathama, Bahu | आजिजिषिष्यन्त , अविवीषिष्यन्त | आजिवयिषिष्यन्त , अविवीषिष्यन्त
| aja~, BvAdi, 262, san, Karmani, Lrn, Madhyama, Eka | आजिजिषिष्यथाः , अविवीषिष्यथाः | आजिवयिषिष्यथाः , अविवीषिष्यथाः
| aja~, BvAdi, 262, san, Karmani, Lrn, Madhyama, Dvi | आजिजिषिष्येथाम् , अविवीषिष्येथाम् | आजिवयिषिष्येथाम् , अविवीषिष्येथाम्
| aja~, BvAdi, 262, san, Karmani, Lrn, Madhyama, Bahu | आजिजिषिष्यध्वम् , अविवीषिष्यध्वम् | आजिवयिषिष्यध्वम् , अविवीषिष्यध्वम्
| aja~, BvAdi, 262, san, Karmani, Lrn, Uttama, Eka | आजिजिषिष्ये , अविवीषिष्ये | आजिवयिषिष्ये , अविवीषिष्ये
| aja~, BvAdi, 262, san, Karmani, Lrn, Uttama, Dvi | आजिजिषिष्यावहि , अविवीषिष्यावहि | आजिवयिषिष्यावहि , अविवीषिष्यावहि
| aja~, BvAdi, 262, san, Karmani, Lrn, Uttama, Bahu | आजिजिषिष्यामहि , अविवीषिष्यामहि | आजिवयिषिष्यामहि , अविवीषिष्यामहि
| zi\Du~, divAdi, 89, Ric, Kartari, Lun, Prathama, Eka | असीषधद् , असीषधत् , असीषधत , असीषिधद् , असीषिधत् , असीषिधत | असीषिधद् , असीषिधत् , असीषिधत , अससाधद् , अससाधत् , अससाधत
| zi\Du~, divAdi, 89, Ric, Kartari, Lun, Prathama, Dvi | असीषधताम् , असीषधेताम् , असीषिधताम् , असीषिधेताम् | असीषिधताम् , असीषिधेताम् , अससाधताम् , अससाधेताम्
| zi\Du~, divAdi, 89, Ric, Kartari, Lun, Prathama, Bahu | असीषधन् , असीषधन्त , असीषिधन् , असीषिधन्त | असीषिधन् , असीषिधन्त , अससाधन् , अससाधन्त
| zi\Du~, divAdi, 89, Ric, Kartari, Lun, Madhyama, Eka | असीषधः , असीषधथाः , असीषिधः , असीषिधथाः | असीषिधः , असीषिधथाः , अससाधः , अससाधथाः
| zi\Du~, divAdi, 89, Ric, Kartari, Lun, Madhyama, Dvi | असीषधतम् , असीषधेथाम् , असीषिधतम् , असीषिधेथाम् | असीषिधतम् , असीषिधेथाम् , अससाधतम् , अससाधेथाम्
| zi\Du~, divAdi, 89, Ric, Kartari, Lun, Madhyama, Bahu | असीषधध्वम् , असीषधत , असीषिधध्वम् , असीषिधत | असीषिधध्वम् , असीषिधत , अससाधध्वम् , अससाधत
| zi\Du~, divAdi, 89, Ric, Kartari, Lun, Uttama, Eka | असीषधम् , असीषधे , असीषिधम् , असीषिधे | असीषिधम् , असीषिधे , अससाधम् , अससाधे
| zi\Du~, divAdi, 89, Ric, Kartari, Lun, Uttama, Dvi | असीषधाव , असीषधावहि , असीषिधाव , असीषिधावहि | असीषिधाव , असीषिधावहि , अससाधाव , अससाधावहि
| zi\Du~, divAdi, 89, Ric, Kartari, Lun, Uttama, Bahu | असीषधाम , असीषधामहि , असीषिधाम , असीषिधामहि | असीषिधाम , असीषिधामहि , अससाधाम , अससाधामहि
| ciY, curAdi, 124, Ric, Karmani, Lun, Prathama, Eka | अचापि , अचायि , अचपि , अचयि | अचपि , अचयि
| brUY, adAdi, 39, yaNluk, Kartari, Lat, Prathama, Eka | बोब्रवीति , बोब्रोति | बोब्रवीति
| brUY, adAdi, 39, yaNluk, Kartari, Lat, Madhyama, Eka | बोब्रवीषि , बोब्रोषि | बोब्रवीषि
| brUY, adAdi, 39, yaNluk, Kartari, Lat, Uttama, Eka | बोब्रवीमि , बोब्रोमि | बोब्रवीमि
| brUY, adAdi, 39, yaNluk, Kartari, Lit, Prathama, Eka | बोब्रवाञ्चकार , बोब्रवामास , बोब्रवाम्बभूव | बोवचाञ्चकार , बोवचामास , बोवचाम्बभूव
| brUY, adAdi, 39, yaNluk, Kartari, Lit, Prathama, Dvi | बोब्रवाञ्चक्रतुः , बोब्रवामासतुः , बोब्रवाम्बभूवतुः | बोवचाञ्चक्रतुः , बोवचामासतुः , बोवचाम्बभूवतुः
| brUY, adAdi, 39, yaNluk, Kartari, Lit, Prathama, Bahu | बोब्रवाञ्चक्रुः , बोब्रवामासुः , बोब्रवाम्बभूवुः | बोवचाञ्चक्रुः , बोवचामासुः , बोवचाम्बभूवुः
| brUY, adAdi, 39, yaNluk, Kartari, Lit, Madhyama, Eka | बोब्रवाञ्चकर्थ , बोब्रवामासिथ , बोब्रवाम्बभूविथ | बोवचाञ्चकर्थ , बोवचामासिथ , बोवचाम्बभूविथ
| brUY, adAdi, 39, yaNluk, Kartari, Lit, Madhyama, Dvi | बोब्रवाञ्चक्रथुः , बोब्रवामासथुः , बोब्रवाम्बभूवथुः | बोवचाञ्चक्रथुः , बोवचामासथुः , बोवचाम्बभूवथुः
| brUY, adAdi, 39, yaNluk, Kartari, Lit, Madhyama, Bahu | बोब्रवाञ्चक्र , बोब्रवामास , बोब्रवाम्बभूव | बोवचाञ्चक्र , बोवचामास , बोवचाम्बभूव
| brUY, adAdi, 39, yaNluk, Kartari, Lit, Uttama, Eka | बोब्रवाञ्चकार , बोब्रवाञ्चकर , बोब्रवामास , बोब्रवाम्बभूव | बोवचाञ्चकार , बोवचाञ्चकर , बोवचामास , बोवचाम्बभूव
| brUY, adAdi, 39, yaNluk, Kartari, Lit, Uttama, Dvi | बोब्रवाञ्चकृव , बोब्रवामासिव , बोब्रवाम्बभूविव | बोवचाञ्चकृव , बोवचामासिव , बोवचाम्बभूविव
| brUY, adAdi, 39, yaNluk, Kartari, Lit, Uttama, Bahu | बोब्रवाञ्चकृम , बोब्रवामासिम , बोब्रवाम्बभूविम | बोवचाञ्चकृम , बोवचामासिम , बोवचाम्बभूविम
| brUY, adAdi, 39, yaNluk, Kartari, Lut, Prathama, Eka | बोब्रविता | बोवचिता
| brUY, adAdi, 39, yaNluk, Kartari, Lut, Prathama, Dvi | बोब्रवितारौ | बोवचितारौ
| brUY, adAdi, 39, yaNluk, Kartari, Lut, Prathama, Bahu | बोब्रवितारः | बोवचितारः
| brUY, adAdi, 39, yaNluk, Kartari, Lut, Madhyama, Eka | बोब्रवितासि | बोवचितासि
| brUY, adAdi, 39, yaNluk, Kartari, Lut, Madhyama, Dvi | बोब्रवितास्थः | बोवचितास्थः
| brUY, adAdi, 39, yaNluk, Kartari, Lut, Madhyama, Bahu | बोब्रवितास्थ | बोवचितास्थ
| brUY, adAdi, 39, yaNluk, Kartari, Lut, Uttama, Eka | बोब्रवितास्मि | बोवचितास्मि
| brUY, adAdi, 39, yaNluk, Kartari, Lut, Uttama, Dvi | बोब्रवितास्वः | बोवचितास्वः
| brUY, adAdi, 39, yaNluk, Kartari, Lut, Uttama, Bahu | बोब्रवितास्मः | बोवचितास्मः
| brUY, adAdi, 39, yaNluk, Kartari, Lrt, Prathama, Eka | बोब्रविष्यति | बोवचिष्यति
| brUY, adAdi, 39, yaNluk, Kartari, Lrt, Prathama, Dvi | बोब्रविष्यतः | बोवचिष्यतः
| brUY, adAdi, 39, yaNluk, Kartari, Lrt, Prathama, Bahu | बोब्रविष्यन्ति | बोवचिष्यन्ति
| brUY, adAdi, 39, yaNluk, Kartari, Lrt, Madhyama, Eka | बोब्रविष्यसि | बोवचिष्यसि
| brUY, adAdi, 39, yaNluk, Kartari, Lrt, Madhyama, Dvi | बोब्रविष्यथः | बोवचिष्यथः
| brUY, adAdi, 39, yaNluk, Kartari, Lrt, Madhyama, Bahu | बोब्रविष्यथ | बोवचिष्यथ
| brUY, adAdi, 39, yaNluk, Kartari, Lrt, Uttama, Eka | बोब्रविष्यामि | बोवचिष्यामि
| brUY, adAdi, 39, yaNluk, Kartari, Lrt, Uttama, Dvi | बोब्रविष्यावः | बोवचिष्यावः
| brUY, adAdi, 39, yaNluk, Kartari, Lrt, Uttama, Bahu | बोब्रविष्यामः | बोवचिष्यामः
| brUY, adAdi, 39, yaNluk, Kartari, Let, Prathama, Eka | बोब्रुवाति | बोवचाति
| brUY, adAdi, 39, yaNluk, Kartari, Let, Prathama, Dvi | बोब्रुवातः | बोवचातः
| brUY, adAdi, 39, yaNluk, Kartari, Let, Prathama, Bahu | बोब्रुवान्ति | बोवचान्ति
| brUY, adAdi, 39, yaNluk, Kartari, Let, Madhyama, Eka | बोब्रुवासि | बोवचासि
| brUY, adAdi, 39, yaNluk, Kartari, Let, Madhyama, Dvi | बोब्रुवाथः | बोवचाथः
| brUY, adAdi, 39, yaNluk, Kartari, Let, Madhyama, Bahu | बोब्रुवाथ | बोवचाथ
| brUY, adAdi, 39, yaNluk, Kartari, Let, Uttama, Eka | बोब्रुवामि | बोवचामि
| brUY, adAdi, 39, yaNluk, Kartari, Let, Uttama, Dvi | बोब्रुवाव , बोब्रुवावः | बोवचाव , बोवचावः
| brUY, adAdi, 39, yaNluk, Kartari, Let, Uttama, Bahu | बोब्रुवाम , बोब्रुवामः | बोवचाम , बोवचामः
| brUY, adAdi, 39, yaNluk, Kartari, Lot, Prathama, Eka | बोब्रूताद् , बोब्रूतात् , बोब्रवीतु , बोब्रोतु | बोब्रूताद् , बोब्रूतात् , बोब्रवीतु
| brUY, adAdi, 39, yaNluk, Kartari, Lan, Prathama, Eka | अबोब्रवीद् , अबोब्रवीत् , अबोब्रोद् , अबोब्रोत् | अबोब्रवीद् , अबोब्रवीत्
| brUY, adAdi, 39, yaNluk, Kartari, Lan, Madhyama, Eka | अबोब्रवीः , अबोब्रोः | अबोब्रवीः
| brUY, adAdi, 39, yaNluk, Kartari, AshirLin, Prathama, Eka | बोब्रूयाद् , बोब्रूयात् | बवुच्याद् , बवुच्यात्
| brUY, adAdi, 39, yaNluk, Kartari, AshirLin, Prathama, Dvi | बोब्रूयास्ताम् | बवुच्यास्ताम्
| brUY, adAdi, 39, yaNluk, Kartari, AshirLin, Prathama, Bahu | बोब्रूयासुः | बवुच्यासुः
| brUY, adAdi, 39, yaNluk, Kartari, AshirLin, Madhyama, Eka | बोब्रूयाः | बवुच्याः
| brUY, adAdi, 39, yaNluk, Kartari, AshirLin, Madhyama, Dvi | बोब्रूयास्तम् | बवुच्यास्तम्
| brUY, adAdi, 39, yaNluk, Kartari, AshirLin, Madhyama, Bahu | बोब्रूयास्त | बवुच्यास्त
| brUY, adAdi, 39, yaNluk, Kartari, AshirLin, Uttama, Eka | बोब्रूयासम् | बवुच्यासम्
| brUY, adAdi, 39, yaNluk, Kartari, AshirLin, Uttama, Dvi | बोब्रूयास्व | बवुच्यास्व
| brUY, adAdi, 39, yaNluk, Kartari, AshirLin, Uttama, Bahu | बोब्रूयास्म | बवुच्यास्म
| brUY, adAdi, 39, yaNluk, Kartari, Lun, Prathama, Eka | अबोब्रावीद् , अबोब्रावीत् | अबोवाचीद् , अबोवाचीत् , अबोवचीद् , अबोवचीत्
| brUY, adAdi, 39, yaNluk, Kartari, Lun, Prathama, Dvi | अबोब्राविष्टाम् | अबोवाचिष्टाम् , अबोवचिष्टाम्
| brUY, adAdi, 39, yaNluk, Kartari, Lun, Prathama, Bahu | अबोब्राविषुः | अबोवाचिषुः , अबोवचिषुः
| brUY, adAdi, 39, yaNluk, Kartari, Lun, Madhyama, Eka | अबोब्रावीः | अबोवाचीः , अबोवचीः
| brUY, adAdi, 39, yaNluk, Kartari, Lun, Madhyama, Dvi | अबोब्राविष्टम् | अबोवाचिष्टम् , अबोवचिष्टम्
| brUY, adAdi, 39, yaNluk, Kartari, Lun, Madhyama, Bahu | अबोब्राविष्ट | अबोवाचिष्ट , अबोवचिष्ट
| brUY, adAdi, 39, yaNluk, Kartari, Lun, Uttama, Eka | अबोब्राविषम् | अबोवाचिषम् , अबोवचिषम्
| brUY, adAdi, 39, yaNluk, Kartari, Lun, Uttama, Dvi | अबोब्राविष्व | अबोवाचिष्व , अबोवचिष्व
| brUY, adAdi, 39, yaNluk, Kartari, Lun, Uttama, Bahu | अबोब्राविष्म | अबोवाचिष्म , अबोवचिष्म
| brUY, adAdi, 39, yaNluk, Kartari, Lrn, Prathama, Eka | अबोब्रविष्यद् , अबोब्रविष्यत् | अबोवचिष्यद् , अबोवचिष्यत्
| brUY, adAdi, 39, yaNluk, Kartari, Lrn, Prathama, Dvi | अबोब्रविष्यताम् | अबोवचिष्यताम्
| brUY, adAdi, 39, yaNluk, Kartari, Lrn, Prathama, Bahu | अबोब्रविष्यन् | अबोवचिष्यन्
| brUY, adAdi, 39, yaNluk, Kartari, Lrn, Madhyama, Eka | अबोब्रविष्यः | अबोवचिष्यः
| brUY, adAdi, 39, yaNluk, Kartari, Lrn, Madhyama, Dvi | अबोब्रविष्यतम् | अबोवचिष्यतम्
| brUY, adAdi, 39, yaNluk, Kartari, Lrn, Madhyama, Bahu | अबोब्रविष्यत | अबोवचिष्यत
| brUY, adAdi, 39, yaNluk, Kartari, Lrn, Uttama, Eka | अबोब्रविष्यम् | अबोवचिष्यम्
| brUY, adAdi, 39, yaNluk, Kartari, Lrn, Uttama, Dvi | अबोब्रविष्याव | अबोवचिष्याव
| brUY, adAdi, 39, yaNluk, Kartari, Lrn, Uttama, Bahu | अबोब्रविष्याम | अबोवचिष्याम
| brUY, adAdi, 39, yaNluk, Karmani, Lat, Prathama, Eka | बोब्रूयते | बवुच्यते
| brUY, adAdi, 39, yaNluk, Karmani, Lat, Prathama, Dvi | बोब्रूयेते | बवुच्येते
| brUY, adAdi, 39, yaNluk, Karmani, Lat, Prathama, Bahu | बोब्रूयन्ते | बवुच्यन्ते
| brUY, adAdi, 39, yaNluk, Karmani, Lat, Madhyama, Eka | बोब्रूयसे | बवुच्यसे
| brUY, adAdi, 39, yaNluk, Karmani, Lat, Madhyama, Dvi | बोब्रूयेथे | बवुच्येथे
| brUY, adAdi, 39, yaNluk, Karmani, Lat, Madhyama, Bahu | बोब्रूयध्वे | बवुच्यध्वे
| brUY, adAdi, 39, yaNluk, Karmani, Lat, Uttama, Eka | बोब्रूये | बवुच्ये
| brUY, adAdi, 39, yaNluk, Karmani, Lat, Uttama, Dvi | बोब्रूयावहे | बवुच्यावहे
| brUY, adAdi, 39, yaNluk, Karmani, Lat, Uttama, Bahu | बोब्रूयामहे | बवुच्यामहे
| brUY, adAdi, 39, yaNluk, Karmani, Lit, Prathama, Eka | बोब्रवाञ्चक्रे , बोब्रवामासे , बोब्रवाम्बभूवे | बोवचाञ्चक्रे , बोवचामासे , बोवचाम्बभूवे
| brUY, adAdi, 39, yaNluk, Karmani, Lit, Prathama, Dvi | बोब्रवाञ्चक्राते , बोब्रवामासाते , बोब्रवाम्बभूवाते | बोवचाञ्चक्राते , बोवचामासाते , बोवचाम्बभूवाते
| brUY, adAdi, 39, yaNluk, Karmani, Lit, Prathama, Bahu | बोब्रवाञ्चक्रिरे , बोब्रवामासिरे , बोब्रवाम्बभूविरे | बोवचाञ्चक्रिरे , बोवचामासिरे , बोवचाम्बभूविरे
| brUY, adAdi, 39, yaNluk, Karmani, Lit, Madhyama, Eka | बोब्रवाञ्चकृषे , बोब्रवामासिषे , बोब्रवाम्बभूविषे | बोवचाञ्चकृषे , बोवचामासिषे , बोवचाम्बभूविषे
| brUY, adAdi, 39, yaNluk, Karmani, Lit, Madhyama, Dvi | बोब्रवाञ्चक्राथे , बोब्रवामासाथे , बोब्रवाम्बभूवाथे | बोवचाञ्चक्राथे , बोवचामासाथे , बोवचाम्बभूवाथे
| brUY, adAdi, 39, yaNluk, Karmani, Lit, Madhyama, Bahu | बोब्रवाञ्चकृढ्वे , बोब्रवामासिध्वे , बोब्रवाम्बभूविढ्वे | बोवचाञ्चकृढ्वे , बोवचामासिध्वे , बोवचाम्बभूविढ्वे
| brUY, adAdi, 39, yaNluk, Karmani, Lit, Uttama, Eka | बोब्रवाञ्चक्रे , बोब्रवामाहे , बोब्रवाम्बभूवे | बोवचाञ्चक्रे , बोवचामाहे , बोवचाम्बभूवे
| brUY, adAdi, 39, yaNluk, Karmani, Lit, Uttama, Dvi | बोब्रवाञ्चकृवहे , बोब्रवामासिवहे , बोब्रवाम्बभूविवहे | बोवचाञ्चकृवहे , बोवचामासिवहे , बोवचाम्बभूविवहे
| brUY, adAdi, 39, yaNluk, Karmani, Lit, Uttama, Bahu | बोब्रवाञ्चकृमहे , बोब्रवामासिमहे , बोब्रवाम्बभूविमहे | बोवचाञ्चकृमहे , बोवचामासिमहे , बोवचाम्बभूविमहे
| brUY, adAdi, 39, yaNluk, Karmani, Lut, Prathama, Eka | बोब्राविता , बोब्रविता | बोवाचिता , बोवचिता
| brUY, adAdi, 39, yaNluk, Karmani, Lut, Prathama, Dvi | बोब्रावितारौ , बोब्रवितारौ | बोवाचितारौ , बोवचितारौ
| brUY, adAdi, 39, yaNluk, Karmani, Lut, Prathama, Bahu | बोब्रावितारः , बोब्रवितारः | बोवाचितारः , बोवचितारः
| brUY, adAdi, 39, yaNluk, Karmani, Lut, Madhyama, Eka | बोब्रावितासे , बोब्रवितासे | बोवाचितासे , बोवचितासे
| brUY, adAdi, 39, yaNluk, Karmani, Lut, Madhyama, Dvi | बोब्रावितासाथे , बोब्रवितासाथे | बोवाचितासाथे , बोवचितासाथे
| brUY, adAdi, 39, yaNluk, Karmani, Lut, Madhyama, Bahu | बोब्राविताध्वे , बोब्रविताध्वे | बोवाचिताध्वे , बोवचिताध्वे
| brUY, adAdi, 39, yaNluk, Karmani, Lut, Uttama, Eka | बोब्राविताहे , बोब्रविताहे | बोवाचिताहे , बोवचिताहे
| brUY, adAdi, 39, yaNluk, Karmani, Lut, Uttama, Dvi | बोब्रावितास्वहे , बोब्रवितास्वहे | बोवाचितास्वहे , बोवचितास्वहे
| brUY, adAdi, 39, yaNluk, Karmani, Lut, Uttama, Bahu | बोब्रावितास्महे , बोब्रवितास्महे | बोवाचितास्महे , बोवचितास्महे
| brUY, adAdi, 39, yaNluk, Karmani, Lrt, Prathama, Eka | बोब्राविष्यते , बोब्रविष्यते | बोवाचिष्यते , बोवचिष्यते
| brUY, adAdi, 39, yaNluk, Karmani, Lrt, Prathama, Dvi | बोब्राविष्येते , बोब्रविष्येते | बोवाचिष्येते , बोवचिष्येते
| brUY, adAdi, 39, yaNluk, Karmani, Lrt, Prathama, Bahu | बोब्राविष्यन्ते , बोब्रविष्यन्ते | बोवाचिष्यन्ते , बोवचिष्यन्ते
| brUY, adAdi, 39, yaNluk, Karmani, Lrt, Madhyama, Eka | बोब्राविष्यसे , बोब्रविष्यसे | बोवाचिष्यसे , बोवचिष्यसे
| brUY, adAdi, 39, yaNluk, Karmani, Lrt, Madhyama, Dvi | बोब्राविष्येथे , बोब्रविष्येथे | बोवाचिष्येथे , बोवचिष्येथे
| brUY, adAdi, 39, yaNluk, Karmani, Lrt, Madhyama, Bahu | बोब्राविष्यध्वे , बोब्रविष्यध्वे | बोवाचिष्यध्वे , बोवचिष्यध्वे
| brUY, adAdi, 39, yaNluk, Karmani, Lrt, Uttama, Eka | बोब्राविष्ये , बोब्रविष्ये | बोवाचिष्ये , बोवचिष्ये
| brUY, adAdi, 39, yaNluk, Karmani, Lrt, Uttama, Dvi | बोब्राविष्यावहे , बोब्रविष्यावहे | बोवाचिष्यावहे , बोवचिष्यावहे
| brUY, adAdi, 39, yaNluk, Karmani, Lrt, Uttama, Bahu | बोब्राविष्यामहे , बोब्रविष्यामहे | बोवाचिष्यामहे , बोवचिष्यामहे
| brUY, adAdi, 39, yaNluk, Karmani, Let, Prathama, Eka | बोब्रूयाते | बवुच्याते
| brUY, adAdi, 39, yaNluk, Karmani, Let, Prathama, Dvi | बोब्रूयैते | बवुच्यैते
| brUY, adAdi, 39, yaNluk, Karmani, Let, Prathama, Bahu | बोब्रूयाते | बवुच्याते
| brUY, adAdi, 39, yaNluk, Karmani, Let, Madhyama, Eka | बोब्रूयासे | बवुच्यासे
| brUY, adAdi, 39, yaNluk, Karmani, Let, Madhyama, Dvi | बोब्रूयैथे | बवुच्यैथे
| brUY, adAdi, 39, yaNluk, Karmani, Let, Madhyama, Bahu | बोब्रूयाध्वे | बवुच्याध्वे
| brUY, adAdi, 39, yaNluk, Karmani, Let, Uttama, Eka | बोब्रूयै | बवुच्यै
| brUY, adAdi, 39, yaNluk, Karmani, Let, Uttama, Dvi | बोब्रूयावहे | बवुच्यावहे
| brUY, adAdi, 39, yaNluk, Karmani, Let, Uttama, Bahu | बोब्रूयामहे | बवुच्यामहे
| brUY, adAdi, 39, yaNluk, Karmani, Lot, Prathama, Eka | बोब्रूयताम् | बवुच्यताम्
| brUY, adAdi, 39, yaNluk, Karmani, Lot, Prathama, Dvi | बोब्रूयेताम् | बवुच्येताम्
| brUY, adAdi, 39, yaNluk, Karmani, Lot, Prathama, Bahu | बोब्रूयन्ताम् | बवुच्यन्ताम्
| brUY, adAdi, 39, yaNluk, Karmani, Lot, Madhyama, Eka | बोब्रूयस्व | बवुच्यस्व
| brUY, adAdi, 39, yaNluk, Karmani, Lot, Madhyama, Dvi | बोब्रूयेथाम् | बवुच्येथाम्
| brUY, adAdi, 39, yaNluk, Karmani, Lot, Madhyama, Bahu | बोब्रूयध्वम् | बवुच्यध्वम्
| brUY, adAdi, 39, yaNluk, Karmani, Lot, Uttama, Eka | बोब्रूयै | बवुच्यै
| brUY, adAdi, 39, yaNluk, Karmani, Lot, Uttama, Dvi | बोब्रूयावहै | बवुच्यावहै
| brUY, adAdi, 39, yaNluk, Karmani, Lot, Uttama, Bahu | बोब्रूयामहै | बवुच्यामहै
| brUY, adAdi, 39, yaNluk, Karmani, Lan, Prathama, Eka | अबोब्रूयत | अबवुच्यत
| brUY, adAdi, 39, yaNluk, Karmani, Lan, Prathama, Dvi | अबोब्रूयेताम् | अबवुच्येताम्
| brUY, adAdi, 39, yaNluk, Karmani, Lan, Prathama, Bahu | अबोब्रूयन्त | अबवुच्यन्त
| brUY, adAdi, 39, yaNluk, Karmani, Lan, Madhyama, Eka | अबोब्रूयथाः | अबवुच्यथाः
| brUY, adAdi, 39, yaNluk, Karmani, Lan, Madhyama, Dvi | अबोब्रूयेथाम् | अबवुच्येथाम्
| brUY, adAdi, 39, yaNluk, Karmani, Lan, Madhyama, Bahu | अबोब्रूयध्वम् | अबवुच्यध्वम्
| brUY, adAdi, 39, yaNluk, Karmani, Lan, Uttama, Eka | अबोब्रूये | अबवुच्ये
| brUY, adAdi, 39, yaNluk, Karmani, Lan, Uttama, Dvi | अबोब्रूयावहि | अबवुच्यावहि
| brUY, adAdi, 39, yaNluk, Karmani, Lan, Uttama, Bahu | अबोब्रूयामहि | अबवुच्यामहि
| brUY, adAdi, 39, yaNluk, Karmani, VidhiLin, Prathama, Eka | बोब्रूयेत | बवुच्येत
| brUY, adAdi, 39, yaNluk, Karmani, VidhiLin, Prathama, Dvi | बोब्रूयेयाताम् | बवुच्येयाताम्
| brUY, adAdi, 39, yaNluk, Karmani, VidhiLin, Prathama, Bahu | बोब्रूयेरन् | बवुच्येरन्
| brUY, adAdi, 39, yaNluk, Karmani, VidhiLin, Madhyama, Eka | बोब्रूयेथाः | बवुच्येथाः
| brUY, adAdi, 39, yaNluk, Karmani, VidhiLin, Madhyama, Dvi | बोब्रूयेयाथाम् | बवुच्येयाथाम्
| brUY, adAdi, 39, yaNluk, Karmani, VidhiLin, Madhyama, Bahu | बोब्रूयेध्वम् | बवुच्येध्वम्
| brUY, adAdi, 39, yaNluk, Karmani, VidhiLin, Uttama, Eka | बोब्रूयेय | बवुच्येय
| brUY, adAdi, 39, yaNluk, Karmani, VidhiLin, Uttama, Dvi | बोब्रूयेवहि | बवुच्येवहि
| brUY, adAdi, 39, yaNluk, Karmani, VidhiLin, Uttama, Bahu | बोब्रूयेमहि | बवुच्येमहि
| brUY, adAdi, 39, yaNluk, Karmani, AshirLin, Prathama, Eka | बोब्राविषीष्ट , बोब्रविषीष्ट | बोवाचिषीष्ट , बोवचिषीष्ट
| brUY, adAdi, 39, yaNluk, Karmani, AshirLin, Prathama, Dvi | बोब्राविषीयास्ताम् , बोब्रविषीयास्ताम् | बोवाचिषीयास्ताम् , बोवचिषीयास्ताम्
| brUY, adAdi, 39, yaNluk, Karmani, AshirLin, Prathama, Bahu | बोब्राविषीरन् , बोब्रविषीरन् | बोवाचिषीरन् , बोवचिषीरन्
| brUY, adAdi, 39, yaNluk, Karmani, AshirLin, Madhyama, Eka | बोब्राविषीष्ठाः , बोब्रविषीष्ठाः | बोवाचिषीष्ठाः , बोवचिषीष्ठाः
| brUY, adAdi, 39, yaNluk, Karmani, AshirLin, Madhyama, Dvi | बोब्राविषीयास्थाम् , बोब्रविषीयास्थाम् | बोवाचिषीयास्थाम् , बोवचिषीयास्थाम्
| brUY, adAdi, 39, yaNluk, Karmani, AshirLin, Madhyama, Bahu | बोब्राविषीढ्वम् , बोब्रविषीढ्वम् | बोवाचिषीध्वम् , बोवचिषीध्वम्
| brUY, adAdi, 39, yaNluk, Karmani, AshirLin, Uttama, Eka | बोब्राविषीय , बोब्रविषीय | बोवाचिषीय , बोवचिषीय
| brUY, adAdi, 39, yaNluk, Karmani, AshirLin, Uttama, Dvi | बोब्राविषीवहि , बोब्रविषीवहि | बोवाचिषीवहि , बोवचिषीवहि
| brUY, adAdi, 39, yaNluk, Karmani, AshirLin, Uttama, Bahu | बोब्राविषीमहि , बोब्रविषीमहि | बोवाचिषीमहि , बोवचिषीमहि
| brUY, adAdi, 39, yaNluk, Karmani, Lun, Prathama, Eka | अबोब्रावि | अबोवाचि
| brUY, adAdi, 39, yaNluk, Karmani, Lun, Prathama, Dvi | अबोब्राविषाताम् , अबोब्रविषाताम् | अबोवाचिषाताम् , अबोवचिषाताम्
| brUY, adAdi, 39, yaNluk, Karmani, Lun, Prathama, Bahu | अबोब्राविषत , अबोब्रविषत | अबोवाचिषत , अबोवचिषत
| brUY, adAdi, 39, yaNluk, Karmani, Lun, Madhyama, Eka | अबोब्राविष्ठाः , अबोब्रविष्ठाः | अबोवाचिष्ठाः , अबोवचिष्ठाः
| brUY, adAdi, 39, yaNluk, Karmani, Lun, Madhyama, Dvi | अबोब्राविषाथाम् , अबोब्रविषाथाम् | अबोवाचिषाथाम् , अबोवचिषाथाम्
| brUY, adAdi, 39, yaNluk, Karmani, Lun, Madhyama, Bahu | अबोब्राविढ्वम् , अबोब्रविढ्वम् | अबोवाचिध्वम् , अबोवचिध्वम्
| brUY, adAdi, 39, yaNluk, Karmani, Lun, Uttama, Eka | अबोब्राविषि , अबोब्रविषि | अबोवाचिषि , अबोवचिषि
| brUY, adAdi, 39, yaNluk, Karmani, Lun, Uttama, Dvi | अबोब्राविष्वहि , अबोब्रविष्वहि | अबोवाचिष्वहि , अबोवचिष्वहि
| brUY, adAdi, 39, yaNluk, Karmani, Lun, Uttama, Bahu | अबोब्राविष्महि , अबोब्रविष्महि | अबोवाचिष्महि , अबोवचिष्महि
| brUY, adAdi, 39, yaNluk, Karmani, Lrn, Prathama, Eka | अबोब्राविष्यत , अबोब्रविष्यत | अबोवाचिष्यत , अबोवचिष्यत
| brUY, adAdi, 39, yaNluk, Karmani, Lrn, Prathama, Dvi | अबोब्राविष्येताम् , अबोब्रविष्येताम् | अबोवाचिष्येताम् , अबोवचिष्येताम्
| brUY, adAdi, 39, yaNluk, Karmani, Lrn, Prathama, Bahu | अबोब्राविष्यन्त , अबोब्रविष्यन्त | अबोवाचिष्यन्त , अबोवचिष्यन्त
| brUY, adAdi, 39, yaNluk, Karmani, Lrn, Madhyama, Eka | अबोब्राविष्यथाः , अबोब्रविष्यथाः | अबोवाचिष्यथाः , अबोवचिष्यथाः
| brUY, adAdi, 39, yaNluk, Karmani, Lrn, Madhyama, Dvi | अबोब्राविष्येथाम् , अबोब्रविष्येथाम् | अबोवाचिष्येथाम् , अबोवचिष्येथाम्
| brUY, adAdi, 39, yaNluk, Karmani, Lrn, Madhyama, Bahu | अबोब्राविष्यध्वम् , अबोब्रविष्यध्वम् | अबोवाचिष्यध्वम् , अबोवचिष्यध्वम्
| brUY, adAdi, 39, yaNluk, Karmani, Lrn, Uttama, Eka | अबोब्राविष्ये , अबोब्रविष्ये | अबोवाचिष्ये , अबोवचिष्ये
| brUY, adAdi, 39, yaNluk, Karmani, Lrn, Uttama, Dvi | अबोब्राविष्यावहि , अबोब्रविष्यावहि | अबोवाचिष्यावहि , अबोवचिष्यावहि
| brUY, adAdi, 39, yaNluk, Karmani, Lrn, Uttama, Bahu | अबोब्राविष्यामहि , अबोब्रविष्यामहि | अबोवाचिष्यामहि , अबोवचिष्यामहि
#### Krdantas
|Derivation for | After | Before |
|---------------|---------|-------|
| kamu~\, BvAdi, 511, , , , ra | कम्र | - |
| BAsf~\, BvAdi, 711, , , , varac | भास्वर | - |
| zWA\, BvAdi, 1077, , , , varac | स्थावर | - |
| hisi~, ruDAdi, 19, , , , ra | हिंस्र | - |
| SF, kryAdi, 21, , , , Aru | शरारु | - |
| ciY, curAdi, 124, , , , Ramul | चापम् , चायम् , चपम् , चयम् | चायम् , चपम् , चयम्
| hisi~, curAdi, 366, , , , ra | हिंस्र | - |
| aja~, BvAdi, 262, san, , , ac | अजिजिष , विवीष | अजिवयिष , विवीष
| aja~, BvAdi, 262, san, , , anIyar | अजिजिषणीय , विवीषणीय | अजिवयिषणीय , विवीषणीय
| aja~, BvAdi, 262, san, , , u | अजिजिषु , विवीषु | अजिवयिषु , विवीषु
| aja~, BvAdi, 262, san, , , kamul | अजिजिषम् , विवीषम् | अजिवयिषम् , विवीषम्
| aja~, BvAdi, 262, san, , , kAnac | अजिजिषाञ्चक्राण , अजिजिषामासान , अजिजिषाम्बभूवाण , विवीषाञ्चक्राण , विवीषामासान , विवीषाम्बभूवाण | अजिवयिषाञ्चक्राण , अजिवयिषामासान , अजिवयिषाम्बभूवाण , विवीषाञ्चक्राण , विवीषामासान , विवीषाम्बभूवाण
| aja~, BvAdi, 262, san, , , kelimar | अजिजिषेलिम , विवीषेलिम | अजिवयिषेलिम , विवीषेलिम
| aja~, BvAdi, 262, san, , , kta | अजिजिषित , विवीषित | अजिवयिषित , विवीषित
| aja~, BvAdi, 262, san, , , ktavatu | अजिजिषितवत् , विवीषितवत् | अजिवयिषितवत् , विवीषितवत्
| aja~, BvAdi, 262, san, , , ktic | अजिजिष्टि , विवीष्टि | अजिवयिष्टि , विवीष्टि
| aja~, BvAdi, 262, san, , , ktin | अजिजिष्टि , विवीष्टि | अजिवयिष्टि , विवीष्टि
| aja~, BvAdi, 262, san, , , ktvA | अजिजिष्य , विवीषित्वा | अजिवयिष्य , विवीषित्वा
| aja~, BvAdi, 262, san, , , kvasu | अजिजिषाञ्चकृवस् , अजिजिषामासिवस् , अजिजिषाम्बभूवस् , विवीषाञ्चकृवस् , विवीषामासिवस् , विवीषाम्बभूवस् | अजिवयिषाञ्चकृवस् , अजिवयिषामासिवस् , अजिवयिषाम्बभूवस् , विवीषाञ्चकृवस् , विवीषामासिवस् , विवीषाम्बभूवस्
| aja~, BvAdi, 262, san, , , kvip | अजिजिस् , विवीस् | अजिवयिस् , विवीस्
| aja~, BvAdi, 262, san, , , Kal | अंजिजिष , अञ्जिजिष , वियंवीष | अंजिवयिष , अञ्जिवयिष , वियंवीष
| aja~, BvAdi, 262, san, , , Ga | अजिजिष , विवीष | अजिवयिष , विवीष
| aja~, BvAdi, 262, san, , , GaY | अजिजिष , विवीष | अजिवयिष , विवीष
| aja~, BvAdi, 262, san, , , Ramul | अजिजिषम् , विवीषम् | अजिवयिषम् , विवीषम्
| aja~, BvAdi, 262, san, , , Rini | अजिजिषिन् , विवीषिन् | अजिवयिषिन् , विवीषिन्
| aja~, BvAdi, 262, san, , , Rvul | अजिजिषक , विवीषक | अजिवयिषक , विवीषक
| aja~, BvAdi, 262, san, , , tavya | अजिजिषितव्य , विवीषितव्य | अजिवयिषितव्य , विवीषितव्य
| aja~, BvAdi, 262, san, , , tavyat | अजिजिषितव्य , विवीषितव्य | अजिवयिषितव्य , विवीषितव्य
| aja~, BvAdi, 262, san, , , tumun | अजिजिषितुम् , विवीषितुम् | अजिवयिषितुम् , विवीषितुम्
| aja~, BvAdi, 262, san, , , tfc | अजिजिषितृ , विवीषितृ | अजिवयिषितृ , विवीषितृ
| aja~, BvAdi, 262, san, , , yat | अजिजिष्य , विवीष्य | अजिवयिष्य , विवीष्य
| aja~, BvAdi, 262, san, , , lyuw | अजिजिषण , विवीषण | अजिवयिषण , विवीषण
| aja~, BvAdi, 262, san, , , vun | अजिजिषक , विवीषक | अजिवयिषक , विवीषक
| kamu~\, BvAdi, 511, san, , , ktvA | चिकामयिषित्वा , चिकमिषित्वा | चिकामयिष्य , चिकमिषित्वा
| ciY, curAdi, 124, Ric, , , Ramul | चापम् , चायम् , चपम् , चयम् | चपम् , चयम्
| brUY, adAdi, 39, yaNluk, , , ac | बोब्रव | बोवच
| brUY, adAdi, 39, yaNluk, , , anIyar | बोब्रवणीय | बोवचनीय
| brUY, adAdi, 39, yaNluk, , , kamul | बोब्रुवम् | बवुचम्
| brUY, adAdi, 39, yaNluk, , , kAnac | बोब्रवाञ्चक्राण , बोब्रवामासान , बोब्रवाम्बभूवाण | बोवचाञ्चक्राण , बोवचामासान , बोवचाम्बभूवान
| brUY, adAdi, 39, yaNluk, , , kelimar | बोब्रुवेलिम | बवुचेलिम
| brUY, adAdi, 39, yaNluk, , , kta | बोब्रुवित | बवुचित
| brUY, adAdi, 39, yaNluk, , , ktavatu | बोब्रुवितवत् | बवुचितवत्
| brUY, adAdi, 39, yaNluk, , , ktic | बोब्रूति | बवुक्ति
| brUY, adAdi, 39, yaNluk, , , ktin | बोब्रूति | बवुक्ति
| brUY, adAdi, 39, yaNluk, , , ktvA | बोब्रवित्वा | बोवचित्वा
| brUY, adAdi, 39, yaNluk, , , kvasu | बोब्रवाञ्चकृवस् , बोब्रवामासिवस् , बोब्रवाम्बभूवस् | बोवचाञ्चकृवस् , बोवचामासिवस् , बोवचाम्बभूवस्
| brUY, adAdi, 39, yaNluk, , , kvip | बोब्रू | बवुच्
| brUY, adAdi, 39, yaNluk, , , Kal | बांब्रव , बाम्ब्रव | बांवच
| brUY, adAdi, 39, yaNluk, , , Ga | बोब्रव | बोवक
| brUY, adAdi, 39, yaNluk, , , GaY | बोब्राव | बोवाक
| brUY, adAdi, 39, yaNluk, , , Ramul | बोब्रावम् | बोवाचम्
| brUY, adAdi, 39, yaNluk, , , Rini | बोब्राविन् | बोवाचिन्
| brUY, adAdi, 39, yaNluk, , , Rvul | बोब्रावक | बोवाचक
| brUY, adAdi, 39, yaNluk, , , tavya | बोब्रवितव्य | बोवचितव्य
| brUY, adAdi, 39, yaNluk, , , tavyat | बोब्रवितव्य | बोवचितव्य
| brUY, adAdi, 39, yaNluk, , , tumun | बोब्रवितुम् | बोवचितुम्
| brUY, adAdi, 39, yaNluk, , , tfc | बोब्रवितृ | बोवचितृ
| brUY, adAdi, 39, yaNluk, , , tfn | बोब्रवितृ | बोवचितृ
| brUY, adAdi, 39, yaNluk, , , lyuw | बोब्रवण | बोवचन
| brUY, adAdi, 39, yaNluk, , , vun | बोब्रवक | बोवचक

tbdasap commented:
@akprasad : Moving this from Draft ---> Ready for review

tbdasap commented on vidyut-prakriya/src/core/term.rs:
```diff
@@ -225,6 +225,10 @@ impl Term {
         self.text.bytes().next().map(|x| x as char)
     }
 
+    pub fn u_adi(&self) -> Option<char> {
```
Done

tbdasap commented on vidyut-prakriya/src/dvitva.rs:
```diff
@@ -195,6 +195,18 @@ fn try_dvitva(rule: Code, p: &mut Prakriya, i_dhatu: usize) -> Option<()> {
             abhyasa.set_adi("");
         }
 
+        // For natva with Upsarga, the original FlagNaAdeshadi needs to be carried
```
DOne

tbdasap commented:
> @tbdasap sounds good -- then, I'll try to look before those changes are ready, but I'll wait for full review until they're all in

@akprasad : Thank you for your patience. It took me more time than I realized to get this in.
The code changes are ready for review. I

 have attached the `table of differences` in output to the PR description.

Updating Readme/Devel setup instructions are still Todos.

akprasad commented:
Looking now.

akprasad commented on vidyut-prakriya/www/static/vidyut-prakriya-app.js:
```diff
@@ -409,25 +442,38 @@ const App = () => ({
+        switch (rule.source) {
+            case "ashtadhyayi":
+                text = this.devaNoSvara(this.sutras[rule.code] || "");
+                annotated_text = text;
```
rather than `annotated_text = text; break`, can just do `return text`, likewise for the `case` statements below.

akprasad commented on vidyut-prakriya/reformat.py:
what is this for?

tbdasap replied:
`python reformat script to markdown table format the differences in the output between the main and current WIP branch based on createAllTinnantas and createAllKrdantas` (in PR description).

The reformated output is postable as a gist [For eg. this PRs diff ](https://gist.github.com/tbdasap/adba3ad6f3db140e1ce39d2bc0315558)

akprasad reviewed (commented):
I'm refamiliarizing myself with the project and Rust after a while, so the review is taking a little longer than I thought. I'll chip away at this a bit every day.

akprasad commented on vidyut-prakriya/Makefile:
```diff
@@ -17,6 +17,11 @@ CURRENT_KRDANTA_OUTPUT:=krdantas_$(GIT_BRANCH).txt
+## Used by "regression_test" target for finding diff with "main".
+## Ensure that you use the same OUTPUT_SCHEME on both branches to avoid useless massive diffs
+## Eg. make regression_test OUTPUT_SCHEME=Slp1
+OUTPUT_SCHEME ?= Devanagari
```
oh my mistake, I thought this was for test files.

akprasad commented on vidyut-prakriya/examples/create_all_krdantas.rs:
```diff
@@ -10,6 +10,7 @@
 //!
 //!     cargo run --release --example create_all_krdantas -- --output-scheme Devanagari
 use clap::Parser;
+use itertools::Itertools;
```
what do we use from this trait? the code looks pretty standard, so I can't tell

tbdasap replied:
items.dedup works.. 
will remove this dependency.

akprasad commented:
as a heads up, I'm merging a fix for https://github.com/ambuda-org/vidyut/issues/216 in the next ~24h or so which will introduce major changes to `wasm.rs` and minor changes to the frontend logic (html/js).

akprasad commented on vidyut-prakriya/tests/integration/regressions.rs:
```diff
@@ -344,3 +346,72 @@ fn apombhyat() {
 }
+
+#[test]
+fn ratva_8_4_14() {
```
for regressions, add a brief comment on what specifically this is fixing. ditto for the tests below.

akprasad commented on vidyut-prakriya/tests/integration/kashika_8_4.rs:
```diff
@@ -265,6 +294,164 @@ fn sutra_8_4_23() {
+
+#[test]
+fn sutra_8_4_34() {
+    assert_has_krdanta(&["pra"], &d("BA\\", Adadi), Krt::lyuw, &["praBAna"]);
```
I find it more readable to move reused dhatus into a var so I can see at a glance that tests are all using the same dhatu, e.g.:

```
let bhaa = d("BA\\", Adadi);
// then, use bhaa in the tests
```

akprasad commented on vidyut-prakriya/tests/integration/kashika_3_2.rs:
```diff
@@ -1426,16 +1434,19 @@ fn sutra_3_2_176() {
+    //     The following "pU" and "jU" are strilinga. In the manner krt/basic.rs executes
+    //     try_add_krt succeeds (kvip?) and try_add_strilinga pratyayas never gets executed.
+    //     Commenting the following tests to argue out the right logic.
+    // assert_has_krdanta(&[], &d("pF", Kryadi), Krt::kvip, &["pU"]);
```
sorry, what happens currently? do we derive nothing or do we derive something incorrect?

akprasad commented on vidyut-prakriya/test_utils/src/lib.rs:
why make these all `pub` ? Aren't they visible from tests already?

tbdasap replied:
1. The macro created version is `pub` but it will not take a `tailored` tester. It always uses a `Tester::default()`
2. If one needs a `tailored tester` as in [kashika_6_2](https://github.com/ambuda-org/vidyut/blob/main/vidyut-prakriya/tests/integration/kashika_6_2.rs#L8-L17)  , there isn't a way to pass it. 
Presuming `assert_has_karmadharaya` is marked `pub` due to this.

Will create a new `assert_samasa_with_tester` macro that will accept a `tailored tester` as param.

akprasad commented on vidyut-prakriya/examples/create_all_tinantas.rs:
```diff
@@ -52,6 +53,7 @@ fn create_output_string(
     output_scheme: Scheme,
 ) -> String {
     items.sort();
+    items = items.into_iter().unique().collect();
```
if you're using itertools for uniqueness, you can also call `items.dedup()` after sorting to remove duplicates.

akprasad reviewed (commented):
still chipping away, about halfway through

akprasad commented on vidyut-prakriya/Cargo.toml:
```diff
@@ -18,6 +18,7 @@ serde = { version = "1.0.150", features = ["derive"] }
 wasm-bindgen = { version = "0.2", features = ["serde-serialize"] }
 serde-wasm-bindgen = "0.4"
 console_error_panic_hook = "0.1.7"
+itertools = "0.10.5"
```
see comment elsewhere on potentially removing itertools

tbdasap replied:
Will remove

akprasad commented on vidyut-prakriya/Makefile:
```diff
@@ -30,7 +35,7 @@ build_all:
 
 # Runs all unit tests in the crate.
-unit_tests:
+unit_tests: build_all
```
I believe `nextest` will build the binaries by default?

tbdasap replied:
Accepted

akprasad commented on vidyut-prakriya/src/ardhadhatuka.rs:
```diff
@@ -280,6 +281,15 @@ pub fn run_before_vikarana(
     let dhatu = p.get(i)?;
     let n = p.pratyaya(j)?;
 
+    if dhatu.has_tag(T::Abhyasta) && !dhatu.has_u("ha\\na~") {
```
I think we also have `is_abhyasta()`

akprasad commented on vidyut-prakriya/src/tripadi/pada_8_4.rs:
```diff
@@ -119,7 +155,14 @@ fn try_natva_for_span(
         if y.has_adi('n') && y.has_tag(T::FlagNaAdeshadi) {
-            ip.run_for_char("8.4.14", i_n, "R");
+            let has_krt =
+                ip.p.find_next_where(i_dhatu, |t| t.is_krt() && !t.is_empty()) != None;
```
saw this elsewhere, but `.is_some()` is preferable to `!= None`

akprasad commented on vidyut-prakriya/src/tripadi/pada_8_4.rs:
```diff
@@ -580,7 +663,18 @@ pub fn run(p: &mut Prakriya) {
     let p = ip.into_p();
 
+    // This is an indicator of a 1.4.14 Pada samjna
+    let last_term = p.terms().last().unwrap();
```
why is this moved here?

akprasad commented on vidyut-prakriya/src/samjna.rs:
```diff
@@ -447,8 +450,10 @@ pub fn try_decide_pratipadika(p: &mut Prakriya) -> Option<()> {
-            p.add_tag_at("1.2.46", i, T::Pratipadika);
-        } else if !t.is_dhatu()
+            // Add tag here but print it later at end of prakriya
+            p.get_mut(i)?.add_tag(T::Pratipadika);
```
why?

tbdasap replied:
This is from a "learnability"/"teachability" perspective (without perturbing the existing program flow).

- 1.4.14 being the "printed" in the last step of the prakriya for subantas/tinnantas.
- 1.2.46 being "printed" for Pratipadika samjna
- 3.1.32  "sanadi" Atideshika Dhatu's it makes sense to "print" it when "prepare Dhatu" completes (and not when the rule gets applied in the program)

akprasad commented on vidyut-prakriya/src/samjna.rs:
```diff
@@ -409,9 +412,9 @@ fn try_run_for_dhatu_pratyaya(p: &mut Prakriya, i: usize) -> Option<()> {
+        if pratyaya.has_lakara(Lit) && !pratyaya.is_ardhadhatuka() {
             p.add_tag_at("3.4.115", i, T::Ardhadhatuka);
-        } else if pratyaya.has_lakara(AshirLin) {
+        } else if pratyaya.has_lakara(AshirLin) && !pratyaya.is_ardhadhatuka() {
```
lit and ashirlin are always ardhadhatuka -- or is the issue that we've run these rules already?

akprasad commented on vidyut-prakriya/src/samjna.rs:
```diff
@@ -243,7 +245,8 @@ pub fn try_run_for_pada_or_bha(p: &mut Prakriya) -> Option<()> {
             } else {
-                p.add_tag_at("1.4.14", i, T::Pada);
+                // Trigger 1.4.14 processing by adding Pada tag but display it in prakriya end
+                p.get_mut(i).unwrap().add_tag(T::Pada);
```
why?

akprasad commented on vidyut-prakriya/src/samjna.rs:
```diff
@@ -70,7 +70,9 @@ pub fn try_pragrhya_rules(p: &mut Prakriya) -> Option<()> {
 
 pub fn try_avyaya_rules(p: &mut Prakriya, i: usize) -> Option<()> {
     let t = p.get(i)?;
```
`let t = p.get_if(i, |t| t.is_avyaya())?`

akprasad reviewed (commented):
A bit more progress -- sorry, it's a complex PR and my schedule is fractured

tbdasap commented on vidyut-prakriya/src/ashtadhyayi.rs:
```diff
@@ -469,14 +479,16 @@ fn run_main_rules(p: &mut Prakriya, dhatu_args: Option<&Dhatu>, args: MainArgs)
-    ardhadhatuka::run_before_vikarana(p, dhatu_args, is_ardhadhatuka, is_lun, lakara);
-    vikarana::run(p);
-    samjna::run(p);
+    if p.find_first_with_tag(Tag::Dhatu) != None {
```
Accepted

tbdasap commented on vidyut-prakriya/tests/integration/kashika_8_4.rs:
```diff
@@ -6,7 +6,7 @@ use vidyut_prakriya::args::Lakara::*;
 use vidyut_prakriya::args::Taddhita as T;
 use vidyut_prakriya::args::*;
-use vidyut_prakriya::Vyakarana;
+pub use vidyut_prakriya::{Decision, Rule, RuleChoice, Vyakarana};
```
Yes: Accepted

tbdasap commented on vidyut-prakriya/tests/integration/regressions.rs:
```diff
@@ -344,3 +346,72 @@ fn apombhyat() {
+#[test]
+#[allow(non_snake_case)]
+fn ratva_8_4_29_1() {
+    // Some examples from Kos#[test]
```
Will fix. examples from kosha actually.

tbdasap commented on vidyut-prakriya/tests/integration/kaumudi_56.rs:
```diff
@@ -124,7 +124,8 @@ fn sk_2651() {
     assert_has_tip(&[], &han, VidhiLin, &["jaNGanyAt"]);
-    // TODO: not sure why vaDyAt is the form here.
+    // To see why vaDyAt is the form here
+    // Refer: https://sa.wikisource.org/wiki/%E0%A4%AA%E0%A5%83%E0%A4%B7%E0%A5%8D%E0%A4%A0%E0%A4%AE%E0%A5%8D%3A%E0%A4%B8%E0%A4%BF%E0%A4%A6%E0%A5%8D%E0%A4%A7%E0%A4%BE%E0%A4%A8%E0%A5%8D%E0%A4%A4%E0%A4%95%E0%A5%8C%E0%A4%AE%E0%A5%81%E0%A4%A6%E0%A5%80_(%E0%A4%AC%E0%A4%BE%E0%A4%B2%E0%A4%AE%E0%A4%A8%E0%A5%8B%E0%A4%B0%E0%A4%AE%E0%A4%BE_%E0%A4%89%E0%A4%A4%E0%A5%8D%E0%A4%A4%E0%A4%B0-%E0%A5%A8).djvu/%E0%A5% ...
```
Accepted: Replaced with a tinyurl and comment

tbdasap commented on vidyut-prakriya/src/dvitva.rs:
```diff
@@ -327,7 +342,24 @@ pub fn try_dvirvacane_aci(p: &mut Prakriya) -> Option<()> {
         let n = p.get(i_n)?;
         if (n.has_adi(AC) && !n.is_it_agama()) || n.has_text("Ji") {
-            run_at_index(p, i);
+            p.debug(format!(
```
Thanks for the guidance.
Will add the #[cfg(debug_assertions)] for those lines..

akprasad commented:
merged in the interest of time, will clean up offline. thank you very much for the changes!
